=== modified file 'src/logic/productionsite.cc'
--- src/logic/productionsite.cc	2011-02-06 12:11:23 +0000
+++ src/logic/productionsite.cc	2011-05-08 13:39:23 +0000
@@ -521,7 +521,7 @@
 	// needs a worker like the one just arrived. That way it is of course still possible, that the worker is
 	// placed on the slot that originally requested the arrived worker.
 	bool worker_placed = false;
-	Ware_Index     idx = widx;
+	Ware_Index     idx = w->worker_index();
 	for (Working_Position * wp = psite.m_working_positions;; ++wp) {
 		if (wp->worker_request == &rq) {
 			if (wp->worker_request->get_index() == idx) {
@@ -532,36 +532,45 @@
 			} else {
 				// Set new request for this slot
 				Ware_Index workerid = wp->worker_request->get_index();
-				delete &rq;
+				delete wp->worker_request;
 				wp->worker_request = &psite.request_worker(workerid);
 			}
 			break;
 		}
 	}
 	while (!worker_placed) {
+		uint8_t nwp = psite.descr().nr_working_positions();
 		uint8_t pos = 0;
-		for (Working_Position * wp = psite.m_working_positions;; ++wp) {
+		for (Working_Position * wp = psite.m_working_positions; pos < nwp; ++wp) {
 			// Find a fitting slot
-			if ((wp->worker_request) && !worker_placed)
+			if (!wp->worker && !worker_placed)
 				if (wp->worker_request->get_index() == idx) {
 					delete wp->worker_request;
 					*wp = Working_Position(0, w);
 					worker_placed = true;
+					break;
 				}
-			pos++;
+			++pos;
 		}
-		// Find the next smaller version of this worker
-		Ware_Index nuwo    = psite.tribe().get_nrworkers();
-		Ware_Index current = Ware_Index(static_cast<size_t>(0));
-		for (; current < nuwo; ++nuwo) {
-			Worker_Descr const * worker = psite.tribe().get_worker_descr(current);
-			if (worker->becomes() == idx) {
-				idx = current;
-				break;
+		if (!worker_placed) {
+			// Find the next smaller version of this worker
+			Ware_Index nuwo    = psite.tribe().get_nrworkers();
+			Ware_Index current = Ware_Index(static_cast<size_t>(0));
+			for (; current < nuwo; ++current) {
+				Worker_Descr const * worker = psite.tribe().get_worker_descr(current);
+				if (worker->becomes() == idx) {
+					idx = current;
+					break;
+				}
 			}
+			if (current == nuwo)
+				throw
+					wexception
+						("Something went wrong! No fitting place for worker %s in %s at (%u, %u) found!",
+						 w->descr().descname().c_str(), psite.descr().descname().c_str(),
+						 psite.get_position().x, psite.get_position().y);
+			pos = 0;
 		}
-		if (current == nuwo)
-			throw wexception("Something went wrong! No fitting place for worker found!");
 	}
 
 	// It's always the first worker doing building work,

