--- CMakeLists.txt	2011-04-04 12:03:24.000000000 +0200
+++ CMakeLists.txt	2011-04-04 23:25:12.000000000 +0200
@@ -337,7 +337,7 @@
 install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/VERSION
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreVersionFile
 )
 
@@ -403,7 +403,7 @@
     pics
     scripting
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreDataFiles
 )
 
@@ -413,7 +413,7 @@
     CREDITS
     ChangeLog
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreLicenseFiles
 )
 
@@ -434,7 +434,7 @@
   DIRECTORY
     ${CMAKE_CURRENT_BINARY_DIR}/locale/
   DESTINATION ${WL_INSTALL_DATADIR}/locale
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreLanguageFiles
 )
 
--- po/CMakeLists.txt	2011-04-04 12:03:24.000000000 +0200
+++ po/CMakeLists.txt	2011-04-04 23:28:46.000000000 +0200
@@ -1,8 +1,8 @@
-if (${CMAKE_BUILD_TYPE} STREQUAL Release)
+if (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
   add_custom_target (
     lang ALL
   )
-else (${CMAKE_BUILD_TYPE} STREQUAL Release)
+else (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
   add_custom_target (
     lang_dummy ALL
     COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/locale
@@ -11,7 +11,7 @@
   add_custom_target (
     lang
   )
-endif (${CMAKE_BUILD_TYPE} STREQUAL Release)
+endif (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
 
 macro(CreateLocale locale_name)
   set (commands "")
