diff -Naur PuzzleMoppetSource-p1/Litha Engine/projects/Puzzle/Events.cpp PuzzleMoppetSource-p2/Litha Engine/projects/Puzzle/Events.cpp
--- PuzzleMoppetSource-p1/Litha Engine/projects/Puzzle/Events.cpp	2012-06-30 13:19:23.000000000 +0200
+++ PuzzleMoppetSource-p2/Litha Engine/projects/Puzzle/Events.cpp	2012-07-13 15:21:09.048032909 +0200
@@ -355,11 +355,11 @@
 			
 			// Emit entire one second's worth of particles all at once.
 			// Then no more...!
-			scene::IParticleEmitter *singleEmissionEmitter = new SingleEmissionEmitter(emitter);
+//			scene::IParticleEmitter *singleEmissionEmitter = new SingleEmissionEmitter(emitter);
 			emitter->drop();
 			
-			ps->setEmitter(singleEmissionEmitter);
-			singleEmissionEmitter->drop();
+//			ps->setEmitter(singleEmissionEmitter);
+//			singleEmissionEmitter->drop();
 			
 			
 			scene::IParticleAffector *affector = ps->createFadeOutParticleAffector();
diff -Naur PuzzleMoppetSource-p1/Litha Engine/source/PostProcessingChain.cpp PuzzleMoppetSource-p2/Litha Engine/source/PostProcessingChain.cpp
--- PuzzleMoppetSource-p1/Litha Engine/source/PostProcessingChain.cpp	2011-09-17 15:06:28.000000000 +0200
+++ PuzzleMoppetSource-p2/Litha Engine/source/PostProcessingChain.cpp	2012-07-13 15:16:01.148032909 +0200
@@ -62,12 +62,12 @@
 	
 	// Create RTT
 	material.TextureLayer[0].Texture = rt;
-	material.TextureLayer[0].TextureWrap = video::ETC_CLAMP_TO_EDGE;
+//	material.TextureLayer[0].TextureWrap = video::ETC_CLAMP_TO_EDGE;
+	
 	
-	/*
 	material.TextureLayer[0].TextureWrapU = video::ETC_CLAMP_TO_EDGE;
 	material.TextureLayer[0].TextureWrapV = video::ETC_CLAMP_TO_EDGE;
-	*/
+	
 	
 	Effect effect =
 	{
