--- FreeOrion/client/AI/CMakeLists.txt~	2010-09-12 22:21:35.000000000 +0600
+++ FreeOrion/client/AI/CMakeLists.txt	2010-09-12 22:21:52.000000000 +0600
@@ -131,7 +131,7 @@
     link_directories(${BOOST_LIBRARYDIR})
     set(THIS_EXE_LINK_LIBS ${THIS_EXE_LINK_LIBS} ${GIGI_GIGI_LIBRARY} ${ZLIB_LIBRARY})
 else ()
-    set(THIS_EXE_LINK_LIBS ${THIS_EXE_LINK_LIBS} ${Boost_LIBRARIES})
+    set(THIS_EXE_LINK_LIBS ${THIS_EXE_LINK_LIBS} ${Boost_LIBRARIES} ${ZLIB_LIBRARY})
 endif ()
 
 executable_all_variants(freeorionca)
