diff -Naur Makefile.linux Makefile.linux
--- Makefile.linux	2008-04-20 12:52:30.000000000 +0200
+++ Makefile.linux	2009-10-31 09:51:42.000000000 +0100
@@ -12,7 +12,7 @@
 SOURCES = $(patsubst %.o, %.c, $(OBJS))
 
 ifeq ($(wildcard .depend),.depend)
-do-it-all: gnp
+do-it-all: gannatsu
 include .depend
 else
 do-it-all: depend recursive
@@ -28,5 +28,5 @@
 clean:
 	$(RM) $(OBJS) *~ .depend gnp.exe
 
-gnp: $(OBJS)
+gannatsu: $(OBJS)
 	$(CXX) -o $@ $(OBJS) $(LDFLAGS)
