diff -Naur act.c act.c
--- act.c	2008-04-23 00:57:20.000000000 +0200
+++ act.c	2009-10-31 11:09:04.000000000 +0100
@@ -321,6 +321,14 @@
 
 void act_init( void )
 {
+#ifdef __LINUX__
+	char workfile[300];
+	sprintf(workfile, "%s/.gannatsu/save/work.sav", getenv("HOME"));
+	char itemfile[300];
+	sprintf(itemfile, "%s/.gannatsu/save/item_wk.sav", getenv("HOME"));
+	char targetfile[300];
+	sprintf(targetfile, "%s/.gannatsu/save/target", getenv("HOME"));
+#endif
 	long i;
 	long x;
 	long y;
@@ -379,7 +387,11 @@
 	}
 	
 	ResetGameFlag2( );
+#ifdef __LINUX__
+	LoadGameFlag2(workfile);
+#else
 	LoadGameFlag2( "save/work.sav");
+#endif
 	for ( i = 0; i < 512; i++ )
 	{
 		save_data[i] = gameflag2[i];
@@ -2220,10 +2232,21 @@
 /* 一時ファイル保存 */
 void save_file_w( )
 {
+#ifdef __LINUX__
+	char configfile[300];
+	sprintf(configfile, "%s/.gannatsu/save/config", getenv("HOME"));
+	char workfile[300];
+	sprintf(workfile, "%s/.gannatsu/save/work.sav", getenv("HOME"));
+	char itemfile[300];
+	sprintf(itemfile, "%s/.gannatsu/save/item_wk.sav", getenv("HOME"));
+#endif
 	long i;
 
+#ifdef __LINUX__
+	SaveGameFlag(configfile);
+#else
 	SaveGameFlag("save/config");
-
+#endif
 	
 	ResetGameFlag2( );
 //	save_data[7] = player[7];	/* ハートのかけら所持個数 */
@@ -2239,16 +2262,22 @@
 	{
 		gameflag2[i] = save_data[i];
 	}
+#ifdef __LINUX__
+	SaveGameFlag2(workfile);
+#else
 	SaveGameFlag2( "save/work.sav");
-	
+#endif
 	/* アイテム取得情報 */
 	ResetGameFlag2();				
 	for ( i = 0; i < 500; i++ )
 	{
 		gameflag2[i] = item_wk[i];
 	}
+#ifdef __LINUX__
+	SaveGameFlag2(itemfile);
+#else
 	SaveGameFlag2("save/item_wk.sav");
-
+#endif
 }
 
 
diff -Naur init.c init.c
--- init.c	2008-04-28 01:52:02.000000000 +0200
+++ init.c	2009-10-31 10:58:54.000000000 +0100
@@ -26,6 +26,17 @@
 
 int main(int argc, char *argv[])
 {
+#ifdef __LINUX__
+	char cfg_path[300];
+	sprintf(cfg_path, "%s/.gannatsu/", getenv("HOME"));
+	mkdir(cfg_path, 0755);
+	char save_path[300];
+	sprintf(save_path, "%s/.gannatsu/save/", getenv("HOME"));
+	mkdir(save_path, 0755);
+	char replay_path[300];
+	sprintf(replay_path, "%s/.gannatsu/replay/", getenv("HOME"));
+	mkdir(replay_path, 0755);
+#endif
 	SDL_Rect rect;
 	Uint32 flags;
 	SDL_Surface* tmp;
@@ -136,12 +147,19 @@
 
 void ExitProgram(void)
 {
-
+#ifdef __LINUX__
+	char configfile[300];
+	sprintf(configfile, "%s/.gannatsu/save/config", getenv("HOME"));
+#endif
 	soundStopBgm(0);
 	soundStopSeAll();
 	soundRelease();
 
-	SaveGameFlag("save/config");
+#ifdef __LINUX__
+	 SaveGameFlag(configfile);
+#else
+	 SaveGameFlag("save/config");
+#endif
 	
 	closePAD();
 //	closeSDL();
@@ -156,11 +174,19 @@
 
 void main_init_config( void )
 {
+#ifdef __LINUX__
+	char configfile[300];
+	sprintf(configfile, "%s/.gannatsu/save/config", getenv("HOME"));
+#endif
 	int i;
 	
 	i = 0;
 	
+#ifdef __LINUX__
+	if ( ! ( LoadGameFlag(configfile) == 0 ) )
+#else
 	if ( ! ( LoadGameFlag("save/config") == 0 ) )
+#endif
 	{
 		/* ファイルが存在しない場合の初期値設定 */
 		ResetGameFlag();
diff -Naur option.c option.c
--- option.c	2008-04-13 19:02:16.000000000 +0200
+++ option.c	2009-10-31 11:11:38.000000000 +0100
@@ -248,6 +248,10 @@
 
 void option_keys( void )
 {
+#ifdef __LINUX__
+	char configfigfile[300];
+	sprintf(configfigfile, "%s/.gannatsu/save/config.fig", getenv("HOME"));
+#endif
 	if ( ag == 0 )
 	{
 		if ( IsPushKey( gameflag[0] ) )
@@ -336,8 +340,11 @@
 		if ( mode == 15 )	/* Exit */
 		{
 			gameflag[133] = selct[2];	//アンカータイプ
+#ifdef __LINUX__
+			SaveGameFlag(configfigfile);
+#else
 			SaveGameFlag("config/config.fig");
-
+#endif
 			gameflag[40] = 1;						/* モードによって変える */
 			g_scene = EN_SN_TITLE;
 			scene_exit=0;
diff -Naur title.c title.c
--- title.c	2008-04-13 19:02:14.000000000 +0200
+++ title.c	2009-10-31 11:13:27.000000000 +0100
@@ -152,6 +152,13 @@
 
 void title_keys( void )
 {
+#ifdef __LINUX__
+	char workfile[300];
+	sprintf(workfile, "%s/.gannatsu/save/work.sav", getenv("HOME"));
+	char itemfile[300];
+	sprintf(itemfile, "%s/.gannatsu/save/item_wk.sav", getenv("HOME"));
+#endif
+
 	if ( IsPushKey( gameflag[0] ) )
 	{
 		soundPlaySe( EN_SE_SELECT );
@@ -315,10 +322,19 @@
 			gameflag2[3]	= 1;	/* 画面Ｎｏ */
 			gameflag2[2]	= 0;	/* ステージ */
 			gameflag[70] = 1;
+#ifdef __LINUX__
+			SaveGameFlag2(workfile);
+#else
 			SaveGameFlag2( "save/work.sav");
+#endif
 
 			ResetGameFlag2( );
+
+#ifdef __LINUX__
+			SaveGameFlag2(itemfile);
+#else
 			SaveGameFlag2( "save/item_wk.sav");
+#endif
 
 			gameflag[40] = 4;
 			g_scene = EN_SN_ACT;
@@ -337,10 +353,19 @@
 				gameflag[123] = gameflag[120];	//リプレイ選択時のステージ
 				gameflag[132] =  1;		//リプレイ
 				gameflag[70] = 1;
+#ifdef __LINUX__
+				SaveGameFlag2(workfile);
+#else
 				SaveGameFlag2( "save/work.sav");
+#endif
 
 				ResetGameFlag2( );
+
+#ifdef __LINUX__
+				SaveGameFlag2(itemfile);
+#else
 				SaveGameFlag2( "save/item_wk.sav");
+#endif
 
 				gameflag[40] = 4;
 				g_scene = EN_SN_ACT;
@@ -362,10 +387,19 @@
 			gameflag[123] = -1;	//リプレイ選択時のステージ
 			gameflag[132] =  0;		//リプレイ
 			gameflag[70] = 1;
+#ifdef __LINUX__
+			SaveGameFlag2(workfile);
+#else
 			SaveGameFlag2( "save/work.sav");
+#endif
 
 			ResetGameFlag2( );
+
+#ifdef __LINUX__
+			SaveGameFlag2(itemfile);
+#else
 			SaveGameFlag2( "save/item_wk.sav");
+#endif
 
 			gameflag[40] = 4;
 			g_scene = EN_SN_ACT;
@@ -387,10 +421,19 @@
 			gameflag[123] = -1;	//リプレイ選択時のステージ
 			gameflag[132] =  0;	//リプレイ
 			gameflag[70] = 1;
+#ifdef __LINUX__
+			SaveGameFlag2(workfile);
+#else
 			SaveGameFlag2( "save/work.sav");
+#endif
 
 			ResetGameFlag2( );
+
+#ifdef __LINUX__
+			SaveGameFlag2(itemfile);
+#else
 			SaveGameFlag2( "save/item_wk.sav");
+#endif
 
 			gameflag[40] = 4;
 			g_scene = EN_SN_ACT;
@@ -434,10 +477,19 @@
 				gameflag[123] = gameflag[120];	//リプレイ選択時のステージ
 				gameflag[132] =  1;		//リプレイ
 				gameflag[70] = 1;
-				SaveGameFlag2( "save/work.sav");
+#ifdef __LINUX__
+			SaveGameFlag2(workfile);
+#else
+			SaveGameFlag2( "save/work.sav");
+#endif
 
-				ResetGameFlag2( );
-				SaveGameFlag2( "save/item_wk.sav");
+			ResetGameFlag2( );
+
+#ifdef __LINUX__
+			SaveGameFlag2(itemfile);
+#else
+			SaveGameFlag2( "save/item_wk.sav");
+#endif
 
 				gameflag[40] = 4;
 				g_scene = EN_SN_ACT;
