diff -Naur atomorun2008-1.0/src/application.d atomorun2008-1.0-p/src/application.d
--- atomorun2008-1.0/src/application.d	2008-03-13 16:33:20.000000000 +0100
+++ atomorun2008-1.0-p/src/application.d	2009-08-28 18:16:49.000000000 +0200
@@ -24,6 +24,8 @@
 
 import std.stdio;
 import std.stream;
+import std.c.stdlib;
+import std.string;
 
 class Application : Task, ActionListener {
 public:
@@ -138,17 +140,28 @@
         };
     };
 
+
+  static const char[] PREF_FILE = ".atomorun2008/savegame";
+
+public char[] pref_file() {
+    char * home = getenv("HOME");
+    if (home is null)
+      throw new Error("HOME environment variable is not defined");
+    return std.string.toString(home) ~ "/" ~ PREF_FILE;
+    }
+
+
 private:
     void addPrize(int p) {
         File savegame = new File();
-        savegame.open("resources/savegame", FileMode.In);
+        savegame.open(pref_file(), FileMode.In);
         int[]   savegameData;
         while (!savegame.eof) {
             savegameData ~= 0;
             savegame.readf(&savegameData[length-1]);
         };
         savegame.close();
-        savegame.open("resources/savegame", FileMode.Out);
+        savegame.open(pref_file(), FileMode.Out);
         if (p>savegameData[0] && attribute == "resources/lvls/lvl1.txt") savegame.writef(p, " ");
             else savegame.writef(savegameData[0], " ");
         if (p>savegameData[1] && attribute == "resources/lvls/lvl2.txt") savegame.writef(p, " ");
diff -Naur atomorun2008-1.0/src/dengine.d atomorun2008-1.0-p/src/dengine.d
--- atomorun2008-1.0/src/dengine.d	2008-03-11 11:44:54.000000000 +0100
+++ atomorun2008-1.0-p/src/dengine.d	2009-08-28 18:17:09.000000000 +0200
@@ -12,7 +12,7 @@
 
 class DEngine : Task {
     void init()     {
-        Config.load("resources/config.cfg");
+        Config.load("/etc/games/atomorun2008/config.cfg");
 
         myTasks ~= Window.getInstance();
         myTasks ~= Input.getInstance();
diff -Naur atomorun2008-1.0/src/menuelvl.d atomorun2008-1.0-p/src/menuelvl.d
--- atomorun2008-1.0/src/menuelvl.d	2008-03-13 16:35:20.000000000 +0100
+++ atomorun2008-1.0-p/src/menuelvl.d	2009-08-28 18:18:20.000000000 +0200
@@ -16,11 +16,23 @@
 import texture2d;
 
 import std.stream;
+import std.c.stdlib;
+import std.string;
 
 import std.stdio;
 
 class MenueLVL : MenueImpl {
 public:
+
+  static const char[] PREF_FILE = ".atomorun2008/savegame";
+
+public char[] pref_file() {
+    char * home = getenv("HOME");
+    if (home is null)
+      throw new Error("HOME environment variable is not defined");
+    return std.string.toString(home) ~ "/" ~ PREF_FILE;
+    }
+
     override void init(Menue theMenue) {
         myButtons.length = 1;
         myButtons[0] = new Button;
@@ -57,7 +69,7 @@
         myImageButtons[6].addActionListener(theMenue, "Game", "resources/lvls/lvl3b.txt");
 
         File hm = new File();
-        hm.open("resources/savegame", FileMode.In);
+        hm.open(pref_file(), FileMode.In);
         int data;
         while (!hm.eof) {
             hm.readf(&data);
