--- CMakeLists.txt	2009-12-14 03:00:09.000000000 +0100
+++ CMakeLists.txt.new	2009-12-14 03:06:33.000000000 +0100
@@ -344,24 +344,45 @@
     INCLUDE(Find3rdPartyDependencies)
 ENDIF()
 
+OPTION(ENABLE_XUL "Enable gecko plugin" ON)
+OPTION(ENABLE_PDF "Enable pdf plugin" ON)
+OPTION(ENABLE_XINE "Enable xine plugin" ON)
+OPTION(ENABLE_JPEG2K "Enable jp2 plugin" ON)
+OPTION(ENABLE_SVG "Enable svg plugin" ON)
+OPTION(ENABLE_FREETYPE "Enable freetype plugin" ON)
+OPTION(ENABLE_CURL "Enable curl plugin" ON)
+OPTION(ENABLE_GIF "Enable gif plugin" ON)
+OPTION(ENABLE_JPEG "Enable jpeg plugin" ON)
+OPTION(ENABLE_PNG "Enable png plugin" ON)
+OPTION(ENABLE_TIFF "Enable tiff plugin" ON)
+OPTION(ENABLE_XRANDR "Enable xrandr usage" ON)
+
 # Common to all platforms:
+IF(ENABLE_FREETYPE)
 FIND_PACKAGE(FreeType)
+ENDIF(ENABLE_FREETYPE)
 FIND_PACKAGE(Inventor)
+IF(ENABLE_JPEG2K)
 FIND_PACKAGE(Jasper)
+ENDIF(ENABLE_JPEG2K)
 FIND_PACKAGE(OpenEXR)
 FIND_PACKAGE(COLLADA)
-FIND_PACKAGE(FBX)
 FIND_PACKAGE(ZLIB)
+IF(ENABLE_XINE)
 FIND_PACKAGE(Xine)
+ENDIF(ENABLE_XINE)
 FIND_PACKAGE(OpenVRML)
 FIND_PACKAGE(Performer)
 FIND_PACKAGE(GDAL)
+IF(ENABLE_CURL)
 FIND_PACKAGE(CURL)
+ENDIF(ENABLE_CURL)
 FIND_PACKAGE(ITK)
 FIND_PACKAGE(LibVNCServer)
 FIND_PACKAGE(OurDCMTK)
-FIND_PACKAGE(OpenAL)
+IF(ENABLE_XUL)
 FIND_PACKAGE(XUL)
+ENDIF(ENABLE_XUL)
 FIND_PACKAGE(FFmpeg)
 FIND_PACKAGE(DirectShow)
 FIND_PACKAGE(SDL)
@@ -381,9 +402,13 @@
         PKG_CHECK_MODULES(GTKGL gtkglext-x11-1.0)
     ENDIF()
 
+    IF(ENABLE_SVG)
     PKG_CHECK_MODULES(RSVG librsvg-2.0)
+    ENDIF(ENABLE_SVG)
     PKG_CHECK_MODULES(CAIRO cairo)
+    IF(ENABLE_PDF)
     PKG_CHECK_MODULES(POPPLER poppler-glib)
+    ENDIF(ENABLE_PDF)
 
 ENDIF()
 
@@ -479,10 +504,18 @@
 # Image readers/writers depend on 3rd party libraries except for OS X which
 # can use Quicktime.
 IF(NOT APPLE)
+    IF(ENABLE_GIF)
     FIND_PACKAGE(GIFLIB)
+    ENDIF(ENABLE_GIF)
+    IF(ENABLE_JPEG)
     FIND_PACKAGE(JPEG)
+    ENDIF(ENABLE_JPEG)
+    IF(ENABLE_PNG)
     FIND_PACKAGE(PNG)
+    ENDIF(ENABLE_PNG)
+    IF(ENABLE_TIFF)
     FIND_PACKAGE(TIFF)
+    ENDIF(ENABLE_TIFF)
 
     # QuickTime is required for OS X, but optional for Windows.
     IF(WIN32)
@@ -919,3 +952,7 @@
   IMMEDIATE @ONLY)
 ADD_CUSTOM_TARGET(uninstall
   "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
+
+MESSAGE(STATUS "<<< Gentoo configuration >>>
+Build type: ${CMAKE_BUILD_TYPE}
+Install path: ${CMAKE_INSTALL_PREFIX}\n")
