--- OpenSceneGraph-3.2.0/src/osgPlugins/freetype/FreeTypeFont.cpp
+++ OpenSceneGraph-3.2.0/src/osgPlugins/freetype/FreeTypeFont.cpp
@@ -14,8 +14,8 @@
 #include "FreeTypeFont.h"
 #include "FreeTypeLibrary.h"
 
-#include <freetype/ftoutln.h>
-#include <freetype/ftbbox.h>
+#include <freetype2/ftoutln.h>
+#include <freetype2/ftbbox.h>
 
 #include <osg/Notify>
 #include <osg/io_utils>
--- OpenSceneGraph-3.2.0/CMakeModules/FindFreeType.cmake
+++ OpenSceneGraph-3.2.0/CMakeModules/FindFreeType.cmake
@@ -48,19 +48,19 @@
   /opt/csw
   /opt
   /usr/freeware
-  PATH_SUFFIXES include
+  PATH_SUFFIXES include include/freetype2
 )
 
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h 
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 config/ftheader.h 
   $ENV{FREETYPE_DIR}/include/freetype2
   NO_DEFAULT_PATH
 )
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h 
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 config/ftheader.h 
   PATHS ${CMAKE_PREFIX_PATH} # Unofficial: We are proposing this.
   NO_DEFAULT_PATH
   PATH_SUFFIXES include/freetype2
 )
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h 
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 config/ftheader.h 
   /usr/local/include/freetype2
   /usr/include/freetype2
   /usr/local/X11R6/include/freetype2
