diff -Naur Bug.cpp Bug.cpp
--- Bug.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Bug.cpp	2009-10-02 08:17:08.000000000 +0200
@@ -1,6 +1,6 @@
 #include "Bug.h"
 #include "Engine.h"
-#include "debris.h"
+#include "Debris.h"
 
 Bug::Bug()
 {
diff -Naur Bullet.cpp Bullet.cpp
--- Bullet.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Bullet.cpp	2009-10-02 08:17:25.000000000 +0200
@@ -1,4 +1,4 @@
-#include "engine.h"
+#include "Engine.h"
 #include "Debris.h"
 #include "RandExplosion.h"
 //================================================================================================//
diff -Naur Editor.cpp Editor.cpp
--- Editor.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Editor.cpp	2009-10-02 08:17:41.000000000 +0200
@@ -1,5 +1,5 @@
 #include "Engine.h"
-#include "UTIL_BitOp.h"
+#include "UTIL_bitop.h"
 //================================================================================================//
 						/************************
 						** Set editor defaults **	
diff -Naur Engine.cpp Engine.cpp
--- Engine.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Engine.cpp	2009-10-02 08:17:54.000000000 +0200
@@ -1,4 +1,4 @@
-#include "engine.h"
+#include "Engine.h"
 //#define PROFILE_TIME 100//set your profile time here (if you want to profile for specific set length)
 //================================================================================================//
 						/***********************
diff -Naur Engine.h Engine.h
--- Engine.h	2009-10-02 08:14:42.000000000 +0200
+++ Engine.h	2009-10-02 08:18:49.000000000 +0200
@@ -2,7 +2,7 @@
 #ifdef WIN32
 #include <windows.h>
 #endif
-#include <sdl.h>
+#include <SDL/SDL.h>
 #include <fstream>
 
 #include "Sound.h"
@@ -49,7 +49,7 @@
 #include "TriggerSlowDown.h"
 #include "Fire.h"
 #include "SolarFlare.h"
-#include "FireBug.h"
+#include "fireBug.h"
 #include "SpaceTrash.h"
 #include <list>
 using namespace std;
diff -Naur Entity.h Entity.h
--- Entity.h	2009-10-02 08:14:42.000000000 +0200
+++ Entity.h	2009-10-02 08:19:12.000000000 +0200
@@ -1,10 +1,10 @@
 #pragma once
 #include "Vector.h"
 #include "Button.h"
-#include "UTIL_Bitop.h"
+#include "UTIL_bitop.h"
 #include "Sprite.h"
 #include "UTIL_Misc.h"
-#include "Geo2d.h"
+#include "Geo2D.h"
 #include "Geo2D_Draw.h"
 #include "FileIO.h"
 #include <list>
diff -Naur firebug.cpp firebug.cpp
--- firebug.cpp	2009-10-02 08:14:42.000000000 +0200
+++ firebug.cpp	2009-10-02 08:22:25.000000000 +0200
@@ -1,4 +1,4 @@
-#include "FireBug.h"
+#include "fireBug.h"
 #include "Engine.h"
 #include "Debris.h"
 FireBug::FireBug()
diff -Naur Map.cpp Map.cpp
--- Map.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Map.cpp	2009-10-02 08:19:25.000000000 +0200
@@ -1,5 +1,5 @@
 #include "Engine.h"
-#include "UTIL_Bitop.h"
+#include "UTIL_bitop.h"
 //================================================================================================//
 						/*********************************
 						** Little cleanup of the arrays **	
diff -Naur PostEffects.cpp PostEffects.cpp
--- PostEffects.cpp	2009-10-02 08:14:42.000000000 +0200
+++ PostEffects.cpp	2009-10-02 08:19:42.000000000 +0200
@@ -1,4 +1,4 @@
-#include "engine.h"
+#include "Engine.h"
 //================================================================================================//
 				/*************************************************************************
 				** if we have pbuffer support, we'll copy the main frame into one image **	
diff -Naur Profiler.h Profiler.h
--- Profiler.h	2009-10-02 08:14:42.000000000 +0200
+++ Profiler.h	2009-10-02 08:19:55.000000000 +0200
@@ -1,5 +1,5 @@
 #pragma
-#include <sdl.h>
+#include <SDL/SDL.h>
 #include <vector>
 #include <string>
 #include <fstream>
diff -Naur Sound.h Sound.h
--- Sound.h	2009-10-02 08:14:42.000000000 +0200
+++ Sound.h	2009-10-02 08:20:12.000000000 +0200
@@ -1,5 +1,5 @@
 #pragma once
-#include <FMOD/fmod.h>
+#include <fmod.h>
 #include <stdio.h>
 
 class Sound
diff -Naur SpriteEffect.cpp SpriteEffect.cpp
--- SpriteEffect.cpp	2009-10-02 08:14:42.000000000 +0200
+++ SpriteEffect.cpp	2009-10-02 08:20:39.000000000 +0200
@@ -1,6 +1,6 @@
 #include "SpriteEffect.h"
 #include "UTIL_openGL.h"
-#include "UTIL_misc.h"
+#include "UTIL_Misc.h"
 #include "Sprite.h"
 #include "Engine.h"
 SpriteEffect::SpriteEffect()
diff -Naur Sprite.h Sprite.h
--- Sprite.h	2009-10-02 08:14:42.000000000 +0200
+++ Sprite.h	2009-10-02 08:20:25.000000000 +0200
@@ -1,5 +1,5 @@
 #pragma once
-#include "UTIL_OpenGL.h"
+#include "UTIL_openGL.h"
 #include "SpriteSheet.h"
 
 void RenderSprite(SpriteSheet spr, int frame, float x, float y, float w, float h);
diff -Naur Timer.h Timer.h
--- Timer.h	2009-10-02 08:14:42.000000000 +0200
+++ Timer.h	2009-10-02 08:20:55.000000000 +0200
@@ -1,5 +1,5 @@
 #pragma once
-#include <sdl.h>
+#include <SDL/SDL.h>
 class Timer
 {
 private:
diff -Naur UTIL_Fade.cpp UTIL_Fade.cpp
--- UTIL_Fade.cpp	2009-10-02 08:14:42.000000000 +0200
+++ UTIL_Fade.cpp	2009-10-02 08:21:13.000000000 +0200
@@ -1,6 +1,6 @@
 #include "UTIL_Fade.h"
 #include "UTIL_Misc.h"
-#include "UTIL_OpenGL.h"
+#include "UTIL_openGL.h"
 
 Fade::Fade()
 {
diff -Naur UTIL_openGL.h UTIL_openGL.h
--- UTIL_openGL.h	2009-10-02 08:14:42.000000000 +0200
+++ UTIL_openGL.h	2009-10-02 08:22:12.000000000 +0200
@@ -3,14 +3,14 @@
 #ifdef _WIN32
 #include <windows.h>
 #endif
-#include <gl/gl.h>
-#include <gl/glu.h>
-#include <gl/glext.h>
-#include <gl/wglext.h>
-#include <il/il.h>
-#include <il/ilu.h>
-#include <il/ilut.h>
-#include <string>
+#include <GL/gl.h>
+#include <GL/glu.h>
+#include <GL/glext.h>
+//#include <gl/wglext.h>
+#include <IL/il.h>
+#include <IL/ilu.h>
+#include <IL/ilut.h>
+#include <cstring>
 #include <sstream>
 #include "Log.h"
 //#define EXTENSION_SUPPORT_1_1	
diff -Naur UTIL_SDL.h UTIL_SDL.h
--- UTIL_SDL.h	2009-10-02 08:14:42.000000000 +0200
+++ UTIL_SDL.h	2009-10-02 08:21:29.000000000 +0200
@@ -1,5 +1,5 @@
 #pragma once
-#include <sdl.h>
+#include <SDL/SDL.h>
 #include "UTIL_openGL.h"
 #include "Log.h"
 #define SDL_VERSION_1_3
