diff -Naur FileIO.h FileIO.h
--- FileIO.h	2009-10-02 08:14:42.000000000 +0200
+++ FileIO.h	2009-10-02 08:36:02.000000000 +0200
@@ -207,4 +207,4 @@
 };
 
 
-#endif _FILEIO_H_
+#endif
diff -Naur Font.h Font.h
--- Font.h	2009-10-02 08:14:42.000000000 +0200
+++ Font.h	2009-10-02 08:36:14.000000000 +0200
@@ -35,7 +35,7 @@
 	/*************************************
 	 ** Methods							**
 	 *************************************/
-	bool Font::GenerateGlyphs(char* name);
+	bool GenerateGlyphs(char* name);
 	void GenerateOffsets(unsigned char* data, int w, int h);//variable widths
 	void GenerateCoords();//texture coordinate per glyph
 	/*************************************
@@ -50,4 +50,4 @@
 	int mAlign;
 	float sine,sineMulti;
 	bool bSineWave;
-};
\ Kein Zeilenumbruch am Dateiende.
+};
diff -Naur FrontEnd.cpp FrontEnd.cpp
--- FrontEnd.cpp	2009-10-02 08:33:58.000000000 +0200
+++ FrontEnd.cpp	2009-10-02 08:36:41.000000000 +0200
@@ -1017,21 +1017,21 @@
 				mFade.StartFadeOut(mTimer.GetTime(),2,3.1f);
 				IntroTimer = mTimer.GetTime()+1;
 				fUIDelay = mTimer.GetTime()+1;
-				if(stricmp(szHiScore[iHiScoreOverWrite],"AAAAA")==0)
+				if(strcmp(szHiScore[iHiScoreOverWrite],"AAAAA")==0)
 				{
 					strcpy(szHiScore[iHiScoreOverWrite],"SHMUP");
 				}
-				if(stricmp(szHiScore[iHiScoreOverWrite],"hyper")==0)
+				if(strcmp(szHiScore[iHiScoreOverWrite],"hyper")==0)
 				{
 					bHyperMode = true;
 					SND_LARGEEXPLODE2;
 				}
-				if(stricmp(szHiScore[iHiScoreOverWrite],"wimps")==0)
+				if(strcmp(szHiScore[iHiScoreOverWrite],"wimps")==0)
 				{
 					bWimpMode = !bWimpMode;
 					SND_LARGEEXPLODE2;
 				}
-				if(stricmp(szHiScore[iHiScoreOverWrite],"bossy")==0)
+				if(strcmp(szHiScore[iHiScoreOverWrite],"bossy")==0)
 				{
 					bBossFight = !bBossFight;
 					SND_LARGEEXPLODE2;
diff -Naur ProtoType.cpp ProtoType.cpp
--- ProtoType.cpp	2009-10-02 08:14:42.000000000 +0200
+++ ProtoType.cpp	2009-10-02 08:36:58.000000000 +0200
@@ -3,7 +3,7 @@
 
 #include "stdafx.h"
 
-int _tmain(int argc, _TCHAR* argv[])
+int main(int argc, char* argv[])
 {
 	Engine engine(640,480,false,"PROTOTYPE");
 
diff -Naur stdafx.h stdafx.h
--- stdafx.h	2009-10-02 08:14:42.000000000 +0200
+++ stdafx.h	2009-10-02 08:37:56.000000000 +0200
@@ -7,7 +7,7 @@
 
 
 #include <iostream>
-#include <tchar.h>
+//#include <tchar.h>
 
 // TODO: reference additional headers your program requires here
 #include "Engine.h"
diff -Naur UTIL_openGL.h UTIL_openGL.h
--- UTIL_openGL.h	2009-10-02 08:22:12.000000000 +0200
+++ UTIL_openGL.h	2009-10-02 08:37:49.000000000 +0200
@@ -1,3 +1,4 @@
+#define ILUT_USE_OPENGL
 #pragma once
 //#include <gl/glew.h>
 #ifdef _WIN32
diff -Naur UTIL_SDL.cpp UTIL_SDL.cpp
--- UTIL_SDL.cpp	2009-10-02 08:14:42.000000000 +0200
+++ UTIL_SDL.cpp	2009-10-02 08:37:32.000000000 +0200
@@ -42,7 +42,7 @@
 		stringstream(str);
 		str << "Resolution Set: " << width << "x" << height << "x" << bpp << endl;
 		gLog.OutPut(str.str());
-		if(!vsync)
+/*		if(!vsync)
 		{
 			PFNWGLSWAPINTERVALEXTPROC wglSwapIntervalEXT  = (PFNWGLSWAPINTERVALEXTPROC) wglGetProcAddress("wglSwapIntervalEXT");
 			if(wglSwapIntervalEXT==NULL)
@@ -50,7 +50,7 @@
 			wglSwapIntervalEXT(0);
 			gLog.OutPut("Vsync Disabled.\n");
 		}
-/*		else
+		else
 		{
 			PFNWGLSWAPINTERVALEXTPROC wglSwapIntervalEXT  = (PFNWGLSWAPINTERVALEXTPROC) wglGetProcAddress("wglSwapIntervalEXT");
 			if(wglSwapIntervalEXT==NULL)
