diff -Naur Engine.cpp Engine.cpp
--- Engine.cpp	2009-10-02 08:27:11.000000000 +0200
+++ Engine.cpp	2009-10-02 08:40:26.000000000 +0200
@@ -20,9 +20,9 @@
 	winHeight = height;
 	int vsync=0, bpp=32, log=1, joystick=1;
 	int fullscreen = fscreen?1:0;
-	int renderTargets=1;
+	int renderTargets=0;
 	int FPS=0;
-	int scanlines = 1;
+	int scanlines = 0;
 	gSerializer.PutComment("Engine.cfg","[Display Properties]");
 /*	gSerializer.ReadVariable("Engine.cfg","winWidth",winWidth);
 	gSerializer.ReadVariable("Engine.cfg","winHeight",winHeight);
@@ -1211,27 +1211,27 @@
 }
 void Engine::SaveScores()
 {
-	DeleteFile("score.dat");
+//	DeleteFile("score.dat");
 	gSerializer.ReadVariable("score.dat","1a",iHiScore[0]);
-	gSerializer.ReadVariable("score.dat","1b",string(szHiScore[0]));
+	gSerializer.ReadVariable("score.dat","1b",iHiScore[0]);
 	gSerializer.ReadVariable("score.dat","2a",iHiScore[1]);
-	gSerializer.ReadVariable("score.dat","2b",string(szHiScore[1]));
+	gSerializer.ReadVariable("score.dat","2b",iHiScore[1]);
 	gSerializer.ReadVariable("score.dat","3a",iHiScore[2]);
-	gSerializer.ReadVariable("score.dat","3b",string(szHiScore[2]));
+	gSerializer.ReadVariable("score.dat","3b",iHiScore[2]);
 	gSerializer.ReadVariable("score.dat","4a",iHiScore[3]);
-	gSerializer.ReadVariable("score.dat","4b",string(szHiScore[3]));
+	gSerializer.ReadVariable("score.dat","4b",iHiScore[3]);
 	gSerializer.ReadVariable("score.dat","5a",iHiScore[4]);
-	gSerializer.ReadVariable("score.dat","5b",string(szHiScore[4]));
+	gSerializer.ReadVariable("score.dat","5b",iHiScore[4]);
 	gSerializer.ReadVariable("score.dat","6a",iHiScore[5]);
-	gSerializer.ReadVariable("score.dat","6b",string(szHiScore[5]));
+	gSerializer.ReadVariable("score.dat","6b",iHiScore[5]);
 	gSerializer.ReadVariable("score.dat","7a",iHiScore[6]);
-	gSerializer.ReadVariable("score.dat","7b",string(szHiScore[6]));
+	gSerializer.ReadVariable("score.dat","7b",iHiScore[6]);
 	gSerializer.ReadVariable("score.dat","8a",iHiScore[7]);
-	gSerializer.ReadVariable("score.dat","8b",string(szHiScore[7]));
+	gSerializer.ReadVariable("score.dat","8b",iHiScore[7]);
 	gSerializer.ReadVariable("score.dat","9a",iHiScore[8]);
-	gSerializer.ReadVariable("score.dat","9b",string(szHiScore[8]));
+	gSerializer.ReadVariable("score.dat","9b",iHiScore[8]);
 	gSerializer.ReadVariable("score.dat","10a",iHiScore[9]);
-	gSerializer.ReadVariable("score.dat","10b",string(szHiScore[9]));
+	gSerializer.ReadVariable("score.dat","10b",iHiScore[9]);
 }
 
 void Engine::RenderScanLines()
diff -Naur FrontEnd.cpp FrontEnd.cpp
--- FrontEnd.cpp	2009-10-02 08:36:41.000000000 +0200
+++ FrontEnd.cpp	2009-10-02 08:40:49.000000000 +0200
@@ -418,7 +418,7 @@
 
 		case 13://back to main
 			//write out new cfg
-			DeleteFile("Controls.cfg");
+//			DeleteFile("Controls.cfg");
 			gSerializer.PutComment("Controls.cfg","[Control Configuration]");
 			for(int n=0; n<6;n++)
 			{
diff -Naur Serializer.cpp Serializer.cpp
--- Serializer.cpp	2009-10-02 08:14:42.000000000 +0200
+++ Serializer.cpp	2009-10-02 08:41:21.000000000 +0200
@@ -46,11 +46,11 @@
 		{
 			getline(f,s);
 			string tok;
-			stringstream(s) >> tok;
+//			stringstream(s) >> tok;
 			if(tok == varname)
 			{
 				string tok2;
-				stringstream(s) >> tok >> tok2 >> value;
+//				stringstream(s) >> tok >> tok2 >> value;
 #ifdef PRINT_SERIAL
 				printf("%s read from file %s\n",tok.c_str(),file.c_str());
 #endif
@@ -84,11 +84,11 @@
 		{
 			getline(f,s);
 			string tok;
-			stringstream(s) >> tok;
+//			stringstream(s) >> tok;
 			if(tok == varname)
 			{
 				string tok2;
-				stringstream(s) >> tok >> tok2 >> value;
+//				stringstream(s) >> tok >> tok2 >> value;
 #ifdef PRINT_SERIAL
 				printf("%s read from file %s\n",tok.c_str(),file.c_str());
 #endif
@@ -123,11 +123,11 @@
 		{
 			getline(f,s);
 			string tok;
-			stringstream(s) >> tok;
+//			stringstream(s) >> tok;
 			if(tok == varname)
 			{
 				string tok2;
-				stringstream(s) >> tok >> tok2 >> value;
+//				stringstream(s) >> tok >> tok2 >> value;
 #ifdef PRINT_SERIAL
 				printf("%s read from file %s\n",tok.c_str(),file.c_str());
 #endif
