diff -urN teeworlds-0.4.3-src.old/default.bam teeworlds-0.4.3-src/default.bam
--- teeworlds-0.4.3-src.old/default.bam	2008-08-31 15:30:59.000000000 +0200
+++ teeworlds-0.4.3-src/default.bam	2008-10-29 18:18:02.000000000 +0100
@@ -207,14 +207,14 @@
 		if platform == "macosx" then
 			settings.cc.flags = "-Wall -fno-exceptions -mmacosx-version-min=10.4"
 		else
-			settings.cc.flags = "-Wall -fstack-protector -fstack-protector-all -fno-exceptions"
+			settings.cc.flags = ""
 		end
 		settings.linker.flags = ""
 	end
 
 	-- set some platform specific settings
 	settings.cc.includes:add("src")
-	settings.cc.includes:add("src/external/zlib")
+	settings.linker.libs:add("z")
 
 	if family == "unix" then
    		if platform == "macosx" then
@@ -244,30 +244,6 @@
 		settings.linker.libs:add("shell32.lib")
 	end
 	
-	-- build glfw
-	glfw_settings = settings:copy()
-	glfw_settings.cc.includes:add("src/external/glfw/include")
-	glfw_settings.cc.includes:add("src/engine/external/glfw/lib")
-	glfw_settings.cc.includes:add("src/engine/external/glfw/lib/" .. glfw_platform)
-
-	glfw = Compile(glfw_settings, Collect(
-		"src/engine/external/glfw/lib/*.c",
-		"src/engine/external/glfw/lib/" .. glfw_platform .. "/*.c"))
-		
-	-- build portaudio
-	pa_settings = settings:copy()
-	pa_settings.cc.defines:add("PA_USE_" .. string.upper(pa_hostapi))
-	pa_settings.cc.defines:add("PA_NO_WMME")
-	pa_settings.cc.defines:add("PA_NO_ASIO")
-	pa_settings.cc.includes:add("src/engine/external/portaudio/include")
-	pa_settings.cc.includes:add("src/engine/external/portaudio/src/common")
-	pa_settings.cc.includes:add("src/engine/external/portaudio/src/os/" .. pa_platform)
-	
-	portaudio = Compile(pa_settings,
-		Collect("src/engine/external/portaudio/src/common/*.c"),
-		Collect("src/engine/external/portaudio/src/os/" .. pa_platform .. "/*.c"),
-		Collect("src/engine/external/portaudio/src/hostapi/" .. pa_hostapi .. "/*.c"))
-	
 	-- build game components
 	engine_settings = settings:copy()
 	
@@ -287,7 +263,7 @@
 
 	-- client
 	client_settings = engine_settings:copy()
-	client_settings.cc.includes:add("src/external/glfw/include")
+	client_settings.cc.includes:add("/usr/include")
 
 	if family == "unix" then
    		if platform == "macosx" then
@@ -303,6 +279,10 @@
 			client_settings.linker.libs:add("X11")
 			client_settings.linker.libs:add("GL")
 			client_settings.linker.libs:add("GLU")
+			client_settings.linker.libs:add("z")
+			client_settings.linker.libs:add("wavpack")
+			client_settings.linker.libs:add("portaudio")
+			client_settings.linker.libs:add("glfw")
 		end
 	elseif family == "windows" then
 		client_settings.linker.libs:add("opengl32.lib")
@@ -311,8 +291,7 @@
 	end
 
 	external_settings = settings:copy()
-	zlib = Compile(external_settings, Collect("src/engine/external/zlib/*.c"))
-	wavpack = Compile(external_settings, Collect("src/engine/external/wavpack/*.c"))
+
 	pnglite = Compile(external_settings, Collect("src/engine/external/pnglite/*.c"))
 
 	engine = Compile(engine_settings, Collect("src/engine/*.c"))
@@ -333,22 +312,22 @@
 	tools = {}
 	for i,v in objs do
 		toolname = PathFilename(PathBase(v))
-		tools[i] = Link(settings, toolname, v, engine, zlib)
+		tools[i] = Link(settings, toolname, v, engine, z)
 	end
 	
 	-- build client, server, version server and master server
 	client_exe = Link(client_settings, "teeworlds", game_shared, game_client,
-		engine, client, game_editor, glfw, portaudio, zlib, pnglite, wavpack,
+		engine, client, game_editor, pnglite, glfw, portaudio, z, wavpack,
 		client_link_other)
 
 	server_exe = Link(server_settings, "teeworlds_srv", engine, server,
-		game_shared, game_server, zlib)
+		game_shared, game_server, z)
 
 	versionserver_exe = Link(server_settings, "versionsrv", versionserver,
-		engine, zlib)
+		engine, z)
 		
 	masterserver_exe = Link(server_settings, "mastersrv", masterserver,
-		engine, zlib)
+		engine, z)
 
 	if platform == "macosx" then
 		osxlaunch_exe = Link(client_settings, "TeeLaunch", osxlaunch)
diff -urN teeworlds-0.4.3-src.old/src/engine/client/ec_gfx.c teeworlds-0.4.3-src/src/engine/client/ec_gfx.c
--- teeworlds-0.4.3-src.old/src/engine/client/ec_gfx.c	2008-08-31 15:30:58.000000000 +0200
+++ teeworlds-0.4.3-src/src/engine/client/ec_gfx.c	2008-10-29 17:51:25.000000000 +0100
@@ -1,5 +1,5 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
-#include <engine/external/glfw/include/GL/glfw.h>
+#include <GL/glfw.h>
 #include <engine/external/pnglite/pnglite.h>
 
 #include <engine/e_system.h>
diff -urN teeworlds-0.4.3-src.old/src/engine/client/ec_inp.c teeworlds-0.4.3-src/src/engine/client/ec_inp.c
--- teeworlds-0.4.3-src.old/src/engine/client/ec_inp.c	2008-08-31 15:30:58.000000000 +0200
+++ teeworlds-0.4.3-src/src/engine/client/ec_inp.c	2008-10-29 17:51:17.000000000 +0100
@@ -1,6 +1,6 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
 #include <string.h>
-#include <engine/external/glfw/include/GL/glfw.h>
+#include <GL/glfw.h>
 
 #include <engine/e_system.h>
 #include <engine/e_client_interface.h>
diff -urN teeworlds-0.4.3-src.old/src/engine/client/ec_snd.c teeworlds-0.4.3-src/src/engine/client/ec_snd.c
--- teeworlds-0.4.3-src.old/src/engine/client/ec_snd.c	2008-08-31 15:30:58.000000000 +0200
+++ teeworlds-0.4.3-src/src/engine/client/ec_snd.c	2008-10-29 18:10:14.000000000 +0100
@@ -3,8 +3,8 @@
 #include <engine/e_client_interface.h>
 #include <engine/e_config.h>
 
-#include <engine/external/portaudio/include/portaudio.h>
-#include <engine/external/wavpack/wavpack.h>
+#include <portaudio.h>
+#include <wavpack/wavpack.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
@@ -406,7 +406,7 @@
 		return -1;
 	snd = &samples[sid];
 
-	context = WavpackOpenFileInput(read_data, error);
+	context = WavpackOpenFileInput(read_data, error, OPEN_TAGS | OPEN_WVC | OPEN_NORMALIZE, 0);
 	if (context)
 	{
 		int samples = WavpackGetNumSamples(context);
diff -urN teeworlds-0.4.3-src.old/src/engine/e_datafile.c teeworlds-0.4.3-src/src/engine/e_datafile.c
--- teeworlds-0.4.3-src.old/src/engine/e_datafile.c	2008-08-31 15:30:58.000000000 +0200
+++ teeworlds-0.4.3-src/src/engine/e_datafile.c	2008-10-29 18:00:11.000000000 +0100
@@ -1,7 +1,7 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
 #include "e_system.h"
 #include "e_datafile.h"
-#include "external/zlib/zlib.h"
+#include <zlib.h>
 
 static const int DEBUG=0;
 
diff -urN teeworlds-0.4.3-src.old/src/engine/external/pnglite/pnglite.c teeworlds-0.4.3-src/src/engine/external/pnglite/pnglite.c
--- teeworlds-0.4.3-src.old/src/engine/external/pnglite/pnglite.c	2008-08-31 15:30:58.000000000 +0200
+++ teeworlds-0.4.3-src/src/engine/external/pnglite/pnglite.c	2008-10-29 18:01:35.000000000 +0100
@@ -5,9 +5,9 @@
 #define USE_ZLIB 1
 
 #if USE_ZLIB
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #else
-#include "zlite.h"
+#include <zlite.h>
 #endif
 
 #include <stdio.h>
