diff -Naur src/Movie_audio.cpp src/Movie_audio.cpp
--- src/Movie_audio.cpp	2012-03-28 19:06:55.689955330 +0200
+++ src/Movie_audio.cpp	2012-03-28 19:09:21.959953697 +0200
@@ -23,7 +23,7 @@
  */
 
 #include "Movie_audio.h"
-#include "Movie.h"
+#include <sfeMovie/Movie.hpp>
 #include <iostream>
 #include <cassert>
 #include "utils.h"
diff -Naur src/Movie.cpp src/Movie.cpp
--- src/Movie.cpp	2012-03-28 19:06:55.689955330 +0200
+++ src/Movie.cpp	2012-03-28 19:09:28.059953629 +0200
@@ -29,7 +29,7 @@
 #include <libswscale/swscale.h>
 }
 
-#include "Movie.h"
+#include <sfeMovie/Movie.hpp>
 #include "Condition.h"
 #include "Movie_video.h"
 #include "Movie_audio.h"
diff -Naur src/Movie_video.cpp src/Movie_video.cpp
--- src/Movie_video.cpp	2012-03-28 19:06:55.689955330 +0200
+++ src/Movie_video.cpp	2012-03-28 19:09:12.779953801 +0200
@@ -24,7 +24,7 @@
 
 
 #include "Movie_video.h"
-#include "Movie.h"
+#include <sfeMovie/Movie.hpp>
 
 #include "Movie_audio.h"
 #include <SFML/Config.hpp>
