--- CMakeLists.txt-o	2012-06-20 16:21:51.000000000 +0200
+++ CMakeLists.txt	2012-06-20 16:23:44.280765606 +0200
@@ -153,12 +153,11 @@
 
 
 if (LINUX)
-	install(DIRECTORY product/include/sfeMovie
-            DESTINATION ${CMAKE_INSTALL_PREFIX}/include
+	INSTALL(FILES include/sfeMovie/Movie.hpp DESTINATION include/sfeMovie/
             COMPONENT devel)
             
-    install(DIRECTORY product/lib
-            DESTINATION ${CMAKE_INSTALL_PREFIX}
+    INSTALL(TARGETS ${LIB_NAME}
+			DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
             COMPONENT devel)
 elseif (MACOSX)
     # edit target properties
