diff -Naur gr_hi/import/opengl.d gr_hi-gentoo/import/opengl.d
--- gr_hi/import/opengl.d	2005-06-18 02:46:00.000000000 +0200
+++ gr_hi-gentoo/import/opengl.d	2007-09-20 08:18:12.000000000 +0200
@@ -1,10 +1,4 @@
-version (Win32) {
-	private import std.c.windows.windows;
-	extern(Windows):
-}
-version (linux) {
 	extern(C):
-}
 
 alias uint GLenum;
 alias ubyte GLboolean;
@@ -1116,7 +1110,7 @@
 /*************************************************************/
 
 void /*APIENTRY*/glAccum (GLenum op, GLfloat value);
-void /*APIENTRY*/glAlphaFunc (GLenum func, GLclampf ref);
+void /*APIENTRY*/glAlphaFunc (GLenum func, GLclampf);
 GLboolean /*APIENTRY*/glAreTexturesResident (GLsizei n, GLuint *textures, GLboolean *residences);
 void /*APIENTRY*/glArrayElement (GLint i);
 void /*APIENTRY*/glBegin (GLenum mode);
@@ -1369,7 +1363,7 @@
 void /*APIENTRY*/glScissor (GLint x, GLint y, GLsizei width, GLsizei height);
 void /*APIENTRY*/glSelectBuffer (GLsizei size, GLuint *buffer);
 void /*APIENTRY*/glShadeModel (GLenum mode);
-void /*APIENTRY*/glStencilFunc (GLenum func, GLint ref, GLuint mask);
+void /*APIENTRY*/glStencilFunc (GLenum func, GLint, GLuint mask);
 void /*APIENTRY*/glStencilMask (GLuint mask);
 void /*APIENTRY*/glStencilOp (GLenum fail, GLenum zfail, GLenum zpass);
 void /*APIENTRY*/glTexCoord1d (GLdouble s);
diff -Naur gr_hi/import/openglu.d gr_hi-gentoo/import/openglu.d
--- gr_hi/import/openglu.d	2005-06-18 02:46:00.000000000 +0200
+++ gr_hi-gentoo/import/openglu.d	2007-09-20 08:15:01.000000000 +0200
@@ -1,11 +1,6 @@
 import opengl;
 
-version (Win32) {
-	extern(Windows):
-}
-version (linux) {
 	extern(C):
-}
 
 GLubyte* gluErrorString (
     GLenum   errCode);
diff -Naur gr_hi/import/SDL.d gr_hi-gentoo/import/SDL.d
--- gr_hi/import/SDL.d	2005-06-18 02:46:00.000000000 +0200
+++ gr_hi-gentoo/import/SDL.d	2007-09-20 08:14:23.000000000 +0200
@@ -20,18 +20,28 @@
     slouken@devolution.com
 */
 
-import SDL_types;
-import SDL_getenv;
-import SDL_error;
-import SDL_rwops;
-import SDL_timer;
-import SDL_audio;
-import SDL_cdrom;
-import SDL_joystick;
-import SDL_events;
-import SDL_video;
-import SDL_byteorder;
-import SDL_Version;
+public import SDL_Keysym;
+public import SDL_Version;
+public import SDL_active;
+public import SDL_audio;
+public import SDL_byteorder;
+public import SDL_cdrom;
+public import SDL_copying;
+public import SDL_endian;
+public import SDL_error;
+public import SDL_events;
+public import SDL_getenv;
+public import SDL_joystick;
+public import SDL_keyboard;
+public import SDL_mouse;
+public import SDL_mutex;
+public import SDL_quit;
+public import SDL_rwops;
+public import SDL_syswm;
+public import SDL_thread;
+public import SDL_timer;
+public import SDL_types;
+public import SDL_video;
 
 extern(C):
 
diff -Naur gr_hi/Makefile gr_hi-gentoo/Makefile
--- gr_hi/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ gr_hi-gentoo/Makefile	2007-09-20 08:14:23.000000000 +0200
@@ -0,0 +1,25 @@
+DC=gdmd
+#DC=gdc
+ifeq ($(DC), gdmd)
+DFLAGS=-O -d -release -Iimport -Isrc
+#DFLAGS=-g -debug -Iimport -Isrc
+DOUT=-of
+else
+DFLAGS=-O -frelease -Iimport -Isrc
+#DFLAGS=-g -fdebug -Iimport -Isrc
+DOUT=-o
+endif
+
+DSRC=$(shell find src/abagames -name "*.d")
+SOURCES=$(DSRC) import/SDL_video.d import/SDL_mixer.d
+OBJS=$(SOURCES:.d=.o)
+EXE=gunroarhi
+
+all: $(EXE)
+
+$(EXE): $(OBJS)
+	gcc -o $@ $(OBJS) -lgphobos -lpthread -lm -lSDL -lGL -lGLU -lSDL_mixer
+
+$(OBJS): %.o: %.d
+	$(DC) -c $(DOUT)$@ $(DFLAGS) $<
+
diff -Naur gr_hi/src/abagames/util/sdl/luminous.d gr_hi-gentoo/src/abagames/util/sdl/luminous.d
--- gr_hi/src/abagames/util/sdl/luminous.d	2005-06-18 02:46:00.000000000 +0200
+++ gr_hi-gentoo/src/abagames/util/sdl/luminous.d	2007-09-20 08:14:23.000000000 +0200
@@ -34,7 +34,7 @@
   private void makeLuminousTexture() {
     uint *data = td;
     int i;
-    memset(data, 0, luminousTextureWidth * luminousTextureHeight * 4 * uint.sizeof);
+  //  memset(data, 0, luminousTextureWidth * luminousTextureHeight * 4 * uint.sizeof);
     glGenTextures(1, &luminousTexture);
     glBindTexture(GL_TEXTURE_2D, luminousTexture);
     glTexImage2D(GL_TEXTURE_2D, 0, 4, luminousTextureWidth, luminousTextureHeight, 0,
diff -Naur gr_hi/src/abagames/util/sdl/mouse.d gr_hi-gentoo/src/abagames/util/sdl/mouse.d
--- gr_hi/src/abagames/util/sdl/mouse.d	2005-09-11 02:47:42.000000000 +0200
+++ gr_hi-gentoo/src/abagames/util/sdl/mouse.d	2007-09-20 08:14:23.000000000 +0200
@@ -11,6 +11,17 @@
 private import abagames.util.sdl.input;
 private import abagames.util.sdl.recordableinput;
 private import abagames.util.sdl.screen;
+
+uint SDL_BUTTON(uint X) { return SDL_PRESSED << (X-1); }
+const uint SDL_BUTTON_LEFT		= 1;
+const uint SDL_BUTTON_MIDDLE	= 2;
+const uint SDL_BUTTON_RIGHT		= 3;
+const uint SDL_BUTTON_WHEELUP	= 4;
+const uint SDL_BUTTON_WHEELDOWN	= 5;
+const uint SDL_BUTTON_LMASK		= SDL_PRESSED << (SDL_BUTTON_LEFT - 1);
+const uint SDL_BUTTON_MMASK		= SDL_PRESSED << (SDL_BUTTON_MIDDLE - 1);
+const uint SDL_BUTTON_RMASK		= SDL_PRESSED << (SDL_BUTTON_RIGHT - 1);
+
 
 /**
  * Mouse input.
