--- d2x-rebirth_v0.58.1-src/SConstruct
+++ d2x-rebirth_v0.58.1-src/SConstruct
@@ -220,7 +220,6 @@
 			env.Append(CPPFLAGS = ['-g'])
 		else:
 			env.Append(CPPDEFINES = ['NDEBUG', 'RELEASE'])
-			env.Append(CPPFLAGS = ['-O2'])
 
 		# profiler?
 		if (self.user_settings.profiler == 1):
@@ -354,6 +353,7 @@
 			self.common_sources += self.sources_use_udp
 
 		env.Append(CPPDEFINES = [('SHAREPATH', '\\"' + str(self.user_settings.sharepath) + '\\"')])
+		env.Append(LINKFLAGS = [str(self.platform_settings.lflags)])
 
 	def _register_program(self,dxxstr,program_specific_objects=[]):
 		env = self.env
@@ -368,7 +368,7 @@
 			versid_cppdefines.append(('DESCENT_VERSION_EXTRA', '\\"%s\\"' % self.user_settings.extra_version))
 		objects.append(self.env.StaticObject(target='%s%s%s' % (self.user_settings.builddir, 'main/vers_id', self.env["OBJSUFFIX"]), source='main/vers_id.c', CPPDEFINES=versid_cppdefines))
 		# finally building program...
-		env.Program(target='%s%s' % (self.user_settings.builddir, str(exe_target)), source = objects, LIBS = self.platform_settings.libs, LINKFLAGS = str(self.platform_settings.lflags))
+		env.Program(target='%s%s' % (self.user_settings.builddir, str(exe_target)), source = objects, LIBS = self.platform_settings.libs)
 		if (sys.platform != 'darwin'):
 			if self.user_settings.register_install_target:
 				install_dir = os.path.join(self.user_settings.DESTDIR or '', self.user_settings.BIN_DIR)
