diff -Naur smw1.7/Makefile smw1.7-p/Makefile
--- smw1.7/Makefile	2007-12-29 21:56:04.000000000 +0100
+++ smw1.7-p/Makefile	2009-09-01 14:33:55.000000000 +0200
@@ -62,7 +62,6 @@
 	rm -rf $(DESTDIR)/usr/share/smw/*/*/.cvsignore
 	rm -rf $(DESTDIR)/usr/share/smw/*/*/*/.cvsignore
 	rm -rf $(DESTDIR)/usr/share/smw/*/*/*/*/.cvsignore
-	chmod a+w $(DESTDIR)/usr/share/smw/maps -R
 
 install-bin : all
 	#assume DESTDIR is the prefix for installing
