diff -Naur RefLectBall/src/gctrl.d RefLectBall-gentoo/src/gctrl.d
--- RefLectBall/src/gctrl.d	2006-10-25 18:56:44.000000000 +0200
+++ RefLectBall-gentoo/src/gctrl.d	2007-10-13 10:20:40.000000000 +0200
@@ -66,6 +66,22 @@
 	GCTRL_RANKING_EXEC,
 	GCTRL_RANKING_EXIT,
 }
+
+public static char[] pref_dir()
+{
+	char * home = getenv("HOME");
+	if (home is null)
+		throw new Error("HOME environment variable not defined");
+	char[] dir = std.string.toString(home) ~ "/.reflectball";
+	char[] dir2 = std.string.toString(home) ~ "/.reflectball/replay";
+	try {
+		mkdir(dir);
+		mkdir(dir2);
+	} catch (FileException e) {
+	}
+	return dir;
+}
+
 
 void TSKgctrl(int id)
 {
@@ -78,7 +94,7 @@
 			replay_data.length = 8;
 		}
 		replay_fmax = 0;
-		fpath = listdir(std.path.curdir ~ "/replay");
+		fpath = listdir(pref_dir() ~ "/" ~ "replay");
 		for(int i = 0; i < fpath.length; i++){
 			if(fnmatch(fpath[i], "rlb??????????????.rep")){
 				replay_file.length = fcnt + 1;
@@ -118,7 +134,7 @@
 				replay_data[i] = replay[i];
 			}
 			char[] fname;
-			fname = format("./replay/rlb%04d%02d%02d%02d%02d%02d.rep"
+			fname = format(pref_dir() ~ "/" ~ "/replay/rlb%04d%02d%02d%02d%02d%02d.rep"
 							,year
 							,month
 							,date
@@ -152,8 +168,8 @@
 	}
 
 	void init_highscore(){
-		if(exists("score.dat")){
-			hi_score = cast(int[])read("score.dat");
+		if(exists(pref_dir() ~ "/" ~ "score.dat")){
+			hi_score = cast(int[])read(pref_dir() ~ "/" ~ "score.dat");
 		}else{
 			hi_score.length = RANKING_MAX * RANKING_RECORD;
 			for(int i = 0; i < RANKING_MAX; i++){
@@ -161,7 +177,7 @@
 				hi_score[i * RANKING_RECORD + RANKING_BREAK] = 0;
 				hi_score[i * RANKING_RECORD + RANKING_DESTR] = 0;
 			}
-			write("score.dat", cast(void[])hi_score);
+			write(pref_dir() ~ "/" ~ "score.dat", cast(void[])hi_score);
 		}
 	}
 
@@ -178,7 +194,7 @@
 				hi_score[i * RANKING_RECORD + RANKING_SCORE] = score;
 				hi_score[i * RANKING_RECORD + RANKING_BREAK] = brk_ship;
 				hi_score[i * RANKING_RECORD + RANKING_DESTR] = enemy_dst;
-				write("score.dat", cast(void[])hi_score);
+				write(pref_dir() ~ "/" ~ "score.dat", cast(void[])hi_score);
 				break;
 			}
 		}
