--- caph-1.1/src/caph.c.orig	2010-10-25 17:40:24.000000000 +0200
+++ caph-1.1/src/caph.c	2011-03-19 00:26:07.127666043 +0100
@@ -37,7 +37,7 @@
 #endif
 
 #define SYS_PRINT	1
-#define SYS_DATA_DIR	"../share/caph/"
+#define SYS_DATA_DIR	"/usr/share/games/caph/"
 
 #define DATA_CONFIG		"caph.conf"
 #define HOME_CONFIG		".caph.conf"
@@ -48,29 +48,8 @@
 static const char *
 sys_get_data_dir()
 {
-	static char dir[256];
-
-#if defined(_WIN32)
-	strcpy(dir, SYS_DATA_DIR);
-	return dir;
-#else
-	int ret = readlink("/proc/self/exe", dir, sizeof(dir));
-	if (ret < 0) {
-		fprintf(stderr, "%s:%i [ERROR] readlink failed: %s \n",
-					__FILE__, __LINE__, strerror(ret));
-		strcpy(dir, SYS_DATA_DIR);
-		return dir;
-	}
-
-	char	*p = dir;
-
-	while (*p != '\0') ++p;
-	while (*p != '/') --p;
-	*++p = '\0';
-
-	strcat(dir, SYS_DATA_DIR);
-	return dir;
-#endif
+	static char dir[FILENAME_MAX];
+	return strcpy(dir, SYS_DATA_DIR);
 }
 
 static void
@@ -79,38 +58,19 @@
 #if defined(_WIN32)
 	SetCurrentDirectory(name);
 #else
-	int	ret;
-
-	ret = chdir(name);
+	chdir(name);
 #endif
 }
 
 static const char *
 sys_get_config()
 {
-	static char dir[256];
-	const char *home;
-
+	static char dir[FILENAME_MAX];
 #if defined(_WIN32)
-	strcpy(dir, DATA_CONFIG);
-	return dir;
+	return strcpy(dir, DATA_CONFIG);
 #else
-	home = getenv("HOME");
-
-	if (home)
-		strcpy(dir, home);
-	else
-		return "";
-
-	char	*p = dir;
-
-	while (*p != '\0') ++p;
-
-	if (*(p - 1) != '/') {
-		*p++ = '/';
-		*p++ = '\0';
-	}
-
+	strcpy(dir, getenv("HOME"));
+	strcat(dir, "/");
 	strcat(dir, HOME_CONFIG);
 	return dir;
 #endif
