diff -Naur z-lock/src/init.d z-lock-p/src/init.d
--- z-lock/src/init.d	2006-09-07 18:32:10.000000000 +0200
+++ z-lock-p/src/init.d	2009-08-28 13:25:55.000000000 +0200
@@ -5,7 +5,8 @@
 
 	2003/12/01 jumpei isshiki
 */
-
+private import  std.c.stdlib;
+private import  std.string;
 private	import	std.stdio;
 private	import	std.stream;
 private	import	std.file;
@@ -16,6 +17,18 @@
 private	import	define;
 private	import	gctrl;
 
+public static char[] pref_dir() {
+  char * home = getenv("HOME");
+  if (home is null)
+    throw new Error("HOME environment variable undefined");
+  char[] dir = std.string.toString(home) ~ "/.zlock";
+  try {
+    mkdir(dir);
+  } catch (FileException e) {
+  }
+  return dir;
+}
+
 void	grpINIT()
 {
 	readSDLtexture("title.bmp", GRP_TITLE);
@@ -80,9 +93,9 @@
 	reptime_mode = 0;
 
 	auto File fd = new File;
-	if(exists("config.dat")){
+	if(exists(pref_dir() ~ "/" ~ "config.dat")){
 		initialized = 0;
-		fd.open("config.dat");
+		fd.open(pref_dir() ~ "/" ~ "config.dat");
 		fd.read(game_ver);
 		fd.read(game_level);
 		fd.read(vol_se);
@@ -96,7 +109,7 @@
 		if(game_ver != GAME_NOWVER) game_ver = GAME_NOWVER;
 	}else{
 		initialized = 1;
-		fd.create("config.dat");
+		fd.create(pref_dir() ~ "/" ~ "config.dat");
 		fd.write(game_ver);
 		fd.write(game_level);
 		fd.write(vol_se);
@@ -137,22 +150,22 @@
 
 	int[] score_tmp;
 
-	if(exists("score.dat")){
-		score_tmp = cast(int[])read("score.dat");
+	if(exists(pref_dir() ~ "/" ~ "score.dat")){
+		score_tmp = cast(int[])read(pref_dir() ~ "/" ~ "score.dat");
 		for(int i = 0; i < high_score.length; i++){
 			for(int j = 0; j < high_score[i].length; j++){
 				high_score[i][j] = score_tmp[i*high_score[i].length+j];
 			}
 		}
 	}else{
-		write("score.dat", cast(void[])high_score);
+		write(pref_dir() ~ "/" ~ "score.dat", cast(void[])high_score);
 	}
 }
 
 void configSAVE()
 {
 	auto File fd = new File;
-	fd.create("config.dat");
+	fd.create(pref_dir() ~ "/" ~ "config.dat");
 	fd.write(game_ver);
 	fd.write(game_level);
 	fd.write(vol_se);
@@ -167,5 +180,5 @@
 
 void scoreSAVE()
 {
-	write("score.dat", cast(void[])high_score);
+	write(pref_dir() ~ "/" ~ "score.dat", cast(void[])high_score);
 }
