diff -Naur z-lock/src/bulletcommand.d z-lock-gentoo/src/bulletcommand.d
--- z-lock/src/bulletcommand.d	2006-09-08 19:28:18.000000000 +0200
+++ z-lock-gentoo/src/bulletcommand.d	2007-10-10 19:40:03.000000000 +0200
@@ -36,7 +36,7 @@
 
 	buf.length = 256;
 	buf = fname ~ "\0";
-	parser[bank] = BulletMLParserTinyXML_new(buf);
+	parser[bank] = BulletMLParserTinyXML_new(buf.ptr);
 	if(parser[bank]) BulletMLParserTinyXML_parse(parser[bank]);
 	buf.length = 0;
 }
@@ -170,7 +170,8 @@
 		if(eid != -1){
 			TskBuf[eid].parent = BulletCommand.now.id;
 			d = (d <= 180.0f ? d : -(360.0f - d));
-			d = d / ROTVAL;
+			d = d / ROTVAL;
+			TskBuf[eid].roll = 0;
 			TskBuf[eid].bullet_speed = s;
 			TskBuf[eid].bullet_direction = d;
 			TskBuf[eid].bullet_velx = (sin(d) * (-s * VEL_SDM_SS_RATIO));
diff -Naur z-lock/src/luminous.d z-lock-gentoo/src/luminous.d
--- z-lock/src/luminous.d	2006-09-07 18:13:08.000000000 +0200
+++ z-lock-gentoo/src/luminous.d	2007-10-10 19:40:44.000000000 +0200
@@ -7,7 +7,7 @@
 */
 
 private	import std.math;
-private	import std.string;
+private	import std.c.string;
 private	import opengl;
 private	import util_sdl;
 private	import task;
@@ -68,11 +68,11 @@
 
 static void makeLuminousTexture()
 {
-	uint *data = td;
+	uint *data = td.ptr;
 	int i;
 
 	td[0..length] = 0;
-	//memset(data, 0, luminousTextureWidth * luminousTextureHeight * 4 * uint.sizeof);
+	memset(data, 0, luminousTextureWidth * luminousTextureHeight * 4 * uint.sizeof);
 	glGenTextures(1, &luminousTexture);
 	glBindTexture(GL_TEXTURE_2D, luminousTexture);
 	glTexImage2D(GL_TEXTURE_2D, 0, 4, luminousTextureWidth, luminousTextureHeight, 0, GL_RGBA, GL_UNSIGNED_BYTE, data);
diff -Naur z-lock/src/main.d z-lock-gentoo/src/main.d
--- z-lock/src/main.d	2005-12-03 14:53:36.000000000 +0100
+++ z-lock-gentoo/src/main.d	2007-09-26 20:22:34.000000000 +0200
@@ -6,13 +6,12 @@
 	2003/11/28 jumpei isshiki
 */
 
-private	import	std.c.windows.windows;
+//private	import	std.c.windows.windows;
 private	import	std.stdio;
 private	import	std.string;
 private	import	std.random;
 private	import	SDL;
 private	import	opengl;
-private	import	reflection;
 private	import	util_sdl;
 private	import	util_glbf;
 private	import	util_pad;
@@ -64,8 +63,8 @@
 				str_buf[i] = lpCmdLine[i];
 			}
 			split_buf = split(str_buf);
-			if(split_buf[0].length > 1) Reflection.init(split_buf[0]);
-			else						Reflection.init("main.exe");
+//			if(split_buf[0].length > 1) Reflection.init(split_buf[0]);
+//			else						Reflection.init("main.exe");
 		}
 
 		try{
@@ -83,7 +82,7 @@
 }else{
 	int main(char[][] argv)
 	{
-		Reflection.init(argv[argv.length-1]);
+//		Reflection.init(argv[argv.length-1]);
 		return boot();
 	}
 }
@@ -154,7 +153,7 @@
 	setTSK(GROUP_00,&TSKgctrl);
 
 	while(game_exec){
-		SDL_PollEvent(&event);
+		SDL_PollEvent (&event);
 		getPAD();
 		if((game_exec == 1 && (trgs & PAD_BUTTON9)) || event.type == SDL_QUIT){
 			game_exec = 0;
@@ -238,7 +237,7 @@
 {
 	int	prev;
 
-	ship_level = 0.0f;
+//	ship_level = 0.0f;
 
 	if(TskBuf[ship_id].tskid != 0 && TskBuf[ship_id].fp_int) collision_sub2(ship_id, GROUP_02);
 	if(TskBuf[ship_id].tskid != 0 && TskBuf[ship_id].fp_int) collision_sub3(ship_id, GROUP_06);
diff -Naur z-lock/src/task.d z-lock-gentoo/src/task.d
--- z-lock/src/task.d	2006-09-08 19:11:52.000000000 +0200
+++ z-lock-gentoo/src/task.d	2007-10-10 19:42:09.000000000 +0200
@@ -135,7 +135,8 @@
 	GROUP_06,
 	GROUP_07,
 	GROUP_08,
-	GROUP_MAX,
+	GROUP_MAX,
+	GROUP_SSHOT = GROUP_03,
 
 }
 
@@ -184,7 +185,9 @@
 		TskBuf[i].fp_exit = null;
 		TskBuf[i].image = null;
 		TskBuf[i].bullet_command = null;
-		TskBuf[i].bullet_state = null;
+		TskBuf[i].bullet_state = null;
+	        TskBuf[i].bullet_accx = 0;
+	        TskBuf[i].bullet_accy = 0;
 	}
 	TskBuf[i].tskid = TSKID_NONE;
 	TskBuf[i].entry = -1;
diff -Naur z-lock/src/title.d z-lock-gentoo/src/title.d
--- z-lock/src/title.d	2006-09-08 18:53:36.000000000 +0200
+++ z-lock-gentoo/src/title.d	2007-09-07 13:00:04.000000000 +0200
@@ -510,7 +510,8 @@
 			str_buf.length = 0;
 			clrTSK(id);
 			break;
-	}
+	}
+	return;
 }
 
 void TSKtitleDraw(int id)
diff -Naur z-lock/src/util_snd.d z-lock-gentoo/src/util_snd.d
--- z-lock/src/util_snd.d	2005-08-24 17:11:26.000000000 +0200
+++ z-lock-gentoo/src/util_snd.d	2007-10-10 19:43:24.000000000 +0200
@@ -99,7 +99,7 @@
 
 	char[] fileName = name ~ "\0";
 
-	music[ch] = Mix_LoadMUS(fileName);
+	music[ch] = Mix_LoadMUS(fileName.ptr);
 	if(!music[ch]){
 		sound_use = false;
 	}else{
@@ -118,7 +118,7 @@
 
 	char[] fileName = name ~ "\0";
 
-	chunk[bank] = Mix_LoadWAV(fileName);
+	chunk[bank] = Mix_LoadWAV(fileName.ptr);
 	if(!chunk[bank]){
 		sound_use = false;
 	}
