diff -Naur src/app.cc src-patch/app.cc
--- src/app.cc	2009-01-21 05:52:46.000000000 +0100
+++ src-patch/app.cc	2009-07-15 12:14:38.000000000 +0200
@@ -1967,11 +1967,13 @@
         else if(!g_iclient) {
           g_iclient.reset(new InputClientLocal());
         }
+#ifdef EXCEPTION_ENABLE_NETPLAY
         else if(InputClientIP* icip = dynamic_cast<InputClientIP*>(g_iclient.get())) {
           if(!icip->isRunning()) {
             throw Error("disconnected");
           }
         }
+#endif // EXCEPTION_ENABLE_NETPLAY
         IGame *game = CreateGame(m_opt);
         CreateGamePanel(game);
 
diff -Naur src/network.cc src-patch/network.cc
--- src/network.cc	2009-01-21 05:52:46.000000000 +0100
+++ src-patch/network.cc	2009-07-15 12:16:36.000000000 +0200
@@ -297,6 +297,7 @@
       ::SetConsoleCtrlHandler(Server_OnShutdown, TRUE);
 #endif // WIN32 
     }
+#ifdef WIN32
     g_print = PrintConsole;
 
     Print(sgui::Format("exception conflict server %.2f\n\n", float(EXCEPTION_VERSION)/100.0f).c_str());
@@ -344,6 +345,7 @@
     if(g_logfile) {
       fclose(g_logfile);
     }
+#endif // WIN32 
   }
 
 
