diff -Naur exception_conflict/src/src/lib/ist/bstream.h exception_conflict-p/src/src/lib/ist/bstream.h
--- exception_conflict/src/src/lib/ist/bstream.h	2009-01-01 19:13:42.000000000 +0100
+++ exception_conflict-p/src/src/lib/ist/bstream.h	2009-08-27 08:34:03.000000000 +0200
@@ -4,9 +4,10 @@
 #define IST_BSTREAM_ENABLE_ZLIB
 
 #include <vector>
-#include <string>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #include <algorithm>
-#include <iostream>
 #include <fstream>
 #ifdef IST_BSTREAM_ENABLE_ZLIB
   #include <zlib.h>
diff -Naur exception_conflict/src/src/lib/sgui/sgui.cc exception_conflict-p/src/src/lib/sgui/sgui.cc
--- exception_conflict/src/src/lib/sgui/sgui.cc	2009-01-16 16:38:20.000000000 +0100
+++ exception_conflict-p/src/src/lib/sgui/sgui.cc	2009-08-27 08:33:22.000000000 +0200
@@ -1,4 +1,7 @@
 #include <algorithm>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #include <SDL/SDL_syswm.h>
 #include "sgui.h"
 
