diff -Naur mars-game-9999/src/Teams/Team.cpp mars-game-9999-p/src/Teams/Team.cpp
--- mars-game-9999/src/Teams/Team.cpp	2011-11-09 17:03:09.150988068 +0100
+++ mars-game-9999-p/src/Teams/Team.cpp	2011-11-09 17:06:37.035985521 +0100
@@ -25,6 +25,7 @@
 # include "Controllers/KeyController.hpp"
 # include "Games/games.hpp"
 # include "Controllers/BotController.hpp"
+#include <set>
 
 Team::Team(Color3f const& color):
     evaluationTimer_(0.f),
