diff --git a/src/BuildTiles/Clipper/Makefile.am b/src/BuildTiles/Clipper/Makefile.am
index 6557c98..eba2694 100644
--- a/src/BuildTiles/Clipper/Makefile.am
+++ b/src/BuildTiles/Clipper/Makefile.am
@@ -5,9 +5,7 @@ dist_pkgdata_DATA = default_priorities.txt
 noinst_PROGRAMS = testclipper
 
 testclipper_SOURCES = testclipper.cxx
-
-testclipper_LDADD = \
-	$(top_builddir)/src/BuildTiles/Clipper/libClipper.a \
+testclipper_LDADD = libClipper.a \
 	$(top_builddir)/src/BuildTiles/Osgb36/libOsgb36.a \
 	$(top_builddir)/src/BuildTiles/Triangulate/libTriangulate.a \
 	$(top_builddir)/src/Lib/Polygon/libPolygon.a \
diff --git a/src/Lib/Array/Makefile.am b/src/Lib/Array/Makefile.am
index 10636d5..acb41db 100644
--- a/src/Lib/Array/Makefile.am
+++ b/src/Lib/Array/Makefile.am
@@ -6,9 +6,7 @@ libArray_a_SOURCES = array.cxx array.hxx
 bin_PROGRAMS = testarray
 
 testarray_SOURCES = testarray.cxx
-
-testarray_LDADD = \
-	$(top_builddir)/src/Lib/Array/libArray.a \
+testarray_LDADD = libArray.a \
 	-lsgbucket -lsgmath -lsgmisc -lsgdebug -lsgxml \
 	$(support_LIBS) -lz 
 
diff --git a/src/Lib/HGT/Makefile.am b/src/Lib/HGT/Makefile.am
index b4ee8cb..16809aa 100644
--- a/src/Lib/HGT/Makefile.am
+++ b/src/Lib/HGT/Makefile.am
@@ -5,9 +5,7 @@ libHGT_a_SOURCES = hgt.cxx hgt.hxx srtmbase.cxx srtmbase.hxx
 noinst_PROGRAMS = testhgt
 
 testhgt_SOURCES = testhgt.cxx
-
-testhgt_LDADD = \
-	$(top_builddir)/src/Lib/HGT/libHGT.a \
+testhgt_LDADD = libHGT.a \
 	-lsgbucket -lsgmisc -lsgdebug -lz -lplibul
 
 INCLUDES = -I$(top_srcdir)/src
