diff -ru src/Airports/GenAirports/Makefile.am src/Airports/GenAirports/Makefile.am
--- src/Airports/GenAirports/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Airports/GenAirports/Makefile.am	2010-07-15 10:05:48.000000000 +0200
@@ -52,7 +52,7 @@
 	$(top_builddir)/src/Lib/Output/libOutput.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgbucket -lsgdebug -lsgio -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip \
+	-lagggpc \
 	-lnewmat \
 	-lz \
 	$(base_LIBS)
diff -ru src/BuildTiles/Clipper/Makefile.am src/BuildTiles/Clipper/Makefile.am
--- src/BuildTiles/Clipper/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/BuildTiles/Clipper/Makefile.am	2010-07-15 10:05:48.000000000 +0200
@@ -16,7 +16,7 @@
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgbucket -lsgdebug -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz 
+	-lagggpc -lz 
 
 INCLUDES = -I$(top_srcdir)/src/Lib -I$(top_srcdir)/src/BuildTiles
 
diff -ru src/BuildTiles/Main/Makefile.am src/BuildTiles/Main/Makefile.am
--- src/BuildTiles/Main/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/BuildTiles/Main/Makefile.am	2010-07-15 10:05:48.000000000 +0200
@@ -21,7 +21,7 @@
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgbucket -lsgio -lsgmath -lsgmisc -lsgdebug -lsgstructure -lsgxml \
-        -lplibsg -lplibul -lgenpolyclip -lz 
+        -lplibsg -lplibul -lagggpc -lz 
 
 fgfs_master_SOURCES = master.cxx
 
diff -ru src/Prep/E00Lines/Makefile.am src/Prep/E00Lines/Makefile.am
--- src/Prep/E00Lines/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/E00Lines/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -10,6 +10,6 @@
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	$(top_builddir)/src/Lib/e00/libe00.a \
 	-lsgdebug -lsgbucket -lsgmisc -lsgmath -lsgio -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
diff -ru src/Prep/GSHHS/Makefile.am src/Prep/GSHHS/Makefile.am
--- src/Prep/GSHHS/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/GSHHS/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -11,7 +11,7 @@
 	$(top_builddir)/src/Lib/Output/libOutput.a \
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	-lsgdebug -lsgbucket -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz 
+	-lagggpc -lz 
 
 debug_SOURCES = \
 	debug.cxx
@@ -21,6 +21,6 @@
 	$(top_builddir)/src/Lib/Geometry/libGeometry.a \
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	-lsgdebug -lsgbucket -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src -I$(top_srcdir)/src/Lib
diff -ru src/Prep/MergerClipper/Makefile.am src/Prep/MergerClipper/Makefile.am
--- src/Prep/MergerClipper/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/MergerClipper/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -13,6 +13,6 @@
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgdebug -lsgbucket -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
diff -ru src/Prep/OGRDecode/Makefile.am src/Prep/OGRDecode/Makefile.am
--- src/Prep/OGRDecode/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/OGRDecode/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -11,6 +11,6 @@
 	$(top_builddir)/src/Lib/poly2tri/libpoly2tri.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgdebug -lsgbucket -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
diff -ru src/Prep/Photo/Makefile.am src/Prep/Photo/Makefile.am
--- src/Prep/Photo/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/Photo/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -37,7 +37,7 @@
 	$(top_builddir)/src/Lib/Output/libOutput.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgbucket -lsgdebug -lsgio -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 	$(base_LIBS)
 
 wgs84offset_SOURCES = wgs84offset.cxx
diff -ru src/Prep/ShapeFile/Makefile.am src/Prep/ShapeFile/Makefile.am
--- src/Prep/ShapeFile/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/ShapeFile/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -12,6 +12,6 @@
 	$(top_builddir)/src/Lib/shapelib/libshape.a \
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	-lsgdebug -lsgbucket -lsgmath -lsgmisc -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
diff -ru src/Prep/TGVPF/Makefile.am src/Prep/TGVPF/Makefile.am
--- src/Prep/TGVPF/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/TGVPF/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -10,6 +10,6 @@
 	$(top_builddir)/src/Lib/TriangleJRS/libTriangleJRS.a \
 	$(top_builddir)/src/Lib/vpf/libvpf.a \
 	-lsgdebug -lsgbucket -lsgmisc -lsgmath -lsgio -lsgtiming -lsgstructure -lsgxml \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
diff -ru src/Prep/UserDef/Makefile.am src/Prep/UserDef/Makefile.am
--- src/Prep/UserDef/Makefile.am	2010-07-15 10:04:49.000000000 +0200
+++ src/Prep/UserDef/Makefile.am	2010-07-15 10:05:47.000000000 +0200
@@ -11,6 +11,6 @@
 	-lsgbucket -lsgmisc -lsgmath -lsgprops -lsgio -lsgstructure -lsgxml \
 	-lsgdebug \
 	-lOpenThreads \
-	-lgenpolyclip -lz
+	-lagggpc -lz
 
 INCLUDES = -I$(top_srcdir)/src/Lib
