--- gigi-9999/GG/src/GIL/extension/io/png_io_private.hpp
+++ gigi-9999/GG/src/GIL/extension/io/png_io_private.hpp
@@ -26,6 +26,10 @@
 #include "../../gil_all.hpp"
 #include "io_error.hpp"
 
+
+#define int_p_NULL                (int *)NULL
+#define png_infopp_NULL           (png_infopp)NULL
+
 namespace boost { namespace gil {
 
 namespace detail {
@@ -312,7 +312,7 @@
             case 4:
             case 8:
                 if(bit_depth < 8)
-                    png_set_gray_1_2_4_to_8(_png_ptr);
+                    png_set_expand_gray_1_2_4_to_8(_png_ptr);
                 png_read_and_convert_pixels<gray8_pixel_t,gray8_ref_t>(view,_cc,_png_ptr,width,height,interlaced);
                 break;
             case 16:
@@ -350,7 +350,7 @@
             case 4:
             case 8:
                 if(bit_depth < 8)
-                    png_set_gray_1_2_4_to_8(_png_ptr);
+                    png_set_expand_gray_1_2_4_to_8(_png_ptr);
                 png_read_and_convert_pixels<gray_alpha8_pixel_t,gray_alpha8_ref_t>(view,_cc,_png_ptr,width,height,interlaced);
                 break;
             case 16:
