--- src/BuildTiles/Main/main.cxx.orig	2009-06-16 16:31:43.000000000 +0200
+++ src/BuildTiles/Main/main.cxx	2009-06-16 16:33:43.000000000 +0200
@@ -1197,12 +1197,12 @@
     // routine can blow up and allocate memory forever.  We'd like
     // this process to die before things get out of hand so we can try
     // again with a smaller interior angle limit.
+    #if 0
     int result;
     struct rlimit limit;
-    limit.rlim_cur = 40000000;
-    limit.rlim_max = 40000000;
+    limit.rlim_cur = 80000000; //rkj doubled value
+    limit.rlim_max = 80000000; //rkj doubled value
 
-#if 0
     result = setrlimit( RLIMIT_DATA, &limit );
     cout << "result of setting mem limit = " << result << endl;
     result = setrlimit( RLIMIT_STACK, &limit );
@@ -1211,14 +1211,14 @@
     cout << "result of setting mem limit = " << result << endl;
     result = setrlimit( RLIMIT_RSS, &limit );
     cout << "result of setting mem limit = " << result << endl;
-#endif
 
     // cpu time limit since occassionally the triangulator can go into
     // an infinite loop.
-    limit.rlim_cur = 300;	// seconds
-    limit.rlim_max = 300;	// seconds
+    limit.rlim_cur = 600;      // seconds rkj doubled value
+    limit.rlim_max = 600;      // seconds rkj doubled value
     result = setrlimit( RLIMIT_CPU, &limit );
     cout << "result of setting mem limit = " << result << endl;
+#endif
 #endif  // end of stuff that crashes Cygwin
 
     // main construction data management class
