diff -Naur openni-9999/Platform/Linux-x86/Build/Modules/nimCodecs/Makefile openni-9999-p/Platform/Linux-x86/Build/Modules/nimCodecs/Makefile
--- openni-9999/Platform/Linux-x86/Build/Modules/nimCodecs/Makefile	2011-11-01 11:08:40.240067646 +0100
+++ openni-9999-p/Platform/Linux-x86/Build/Modules/nimCodecs/Makefile	2011-11-01 11:04:41.000000000 +0100
@@ -2,17 +2,21 @@
 
 INC_DIRS = \
 	../../../../../Include \
-	../../../../../Source \
-	../../../../../Source/External/LibJPEG
+	../../../../../Source
 
 SRC_FILES = \
-	../../../../../Source/Modules/nimCodecs/*.cpp \
-	../../../../../Source/External/LibJPEG/*.c
+	../../../../../Source/Modules/nimCodecs/*.cpp
 
 LIB_NAME = nimCodecs
 USED_LIBS = OpenNI
 
-include ../../Common/CommonCppMakefile
-
+ifneq ($(wildcard /usr/lib/libjpeg.so /usr/lib32/libjpeg.so /usr/lib64/libjpeg.so /usr/local/lib/libjpeg.so /usr/local/lib32/libjpeg.so /usr/local/lib64/libjpeg.so),)
+  USED_LIBS += jpeg
+else
+  INC_DIRS  += ../../../../../Source/External/LibJPEG
+  SRC_FILES += ../../../../../Source/External/LibJPEG/*.c
+endif
+  
+include ../../Common/CommonCppMakefile
 
 
