diff -Naur mytoy-se-1.0.0/ccore.cpp mytoy-se-1.0.0-home/ccore.cpp
--- mytoy-se-1.0.0/ccore.cpp	2008-06-13 18:48:43.000000000 +0200
+++ mytoy-se-1.0.0-home/ccore.cpp	2009-11-24 12:34:51.000000000 +0100
@@ -181,8 +181,48 @@
 // Kamera megnyitása
 // ==================
 int CCore::OpenCamera() {
+	char cfg_path[300];
+	sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+	mkdir(cfg_path, 0755);
+	sprintf(cfg_path, "%s/.mytoyse/configuration", getenv("HOME"));
+	FILE *f = fopen(cfg_path, "r");
 	
-	device = cvCaptureFromCAM( camdesc.number );	
+	if ( !f ) 
+	device = cvCaptureFromCAM( camdesc.number );				 
+	else 		
+		{	
+	 		char line[80];
+			char *token;
+			int  param;
+	while ( fgets(line, 80, f ) ) {
+	
+				// megjegyzés vagy üres sor
+				if ( line[0] == '#' )
+					continue;
+			
+				token = strtok(line, "= \t");
+		
+				param = -1;
+				if ( strstr( token, "video_device" ) ) 		param = 4;
+									
+				token = strtok( NULL, "= \t\n" );
+					
+				switch ( param ) {
+					
+					case -1:
+						break;
+					case 4: 
+						video_device = atoi( token );
+						device = cvCaptureFromCAM( video_device );
+						break;	
+				}
+			}
+			
+			fclose(f);			
+	 	}
+
+
+//	device = cvCaptureFromCAM( camdesc.number );	
 	
 	if (!device) 
 	    return 0;
@@ -361,7 +401,11 @@
 	
 	
 	// config file feldolgozása
-	FILE *f = fopen("configuration", "r");
+	char cfg_path[300];
+	sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+	mkdir(cfg_path, 0755);
+	sprintf(cfg_path, "%s/.mytoyse/configuration", getenv("HOME"));
+	FILE *f = fopen(cfg_path, "r");
 	
 	if ( !f ) 
 		return 1;				 
@@ -591,7 +635,11 @@
 							break;
 								
 					case SDLK_s:
-							ScreenShot("screenshots/TestMode", ++screenshot, 0, 0, GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/TestMode", getenv("HOME"));
+							ScreenShot(cfg_path, ++screenshot, 0, 0, GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;
 					default: ;
 				} 
diff -Naur mytoy-se-1.0.0/ccore.h mytoy-se-1.0.0-home/ccore.h
--- mytoy-se-1.0.0/ccore.h	2005-02-21 21:16:18.000000000 +0100
+++ mytoy-se-1.0.0-home/ccore.h	2009-11-24 12:32:21.000000000 +0100
@@ -101,6 +101,8 @@
 			int					time_stamp_period;		// idõbélyeg
 			int					vertical_mirror;		// legyen-e vertikális/horizontális tükrözés
 			int					horizontal_mirror;
+			int					video_device;
+
 															
 	public:
 			CCore();
diff -Naur mytoy-se-1.0.0/ccrazy.cpp mytoy-se-1.0.0-home/ccrazy.cpp
--- mytoy-se-1.0.0/ccrazy.cpp	2008-06-13 18:59:26.000000000 +0200
+++ mytoy-se-1.0.0-home/ccrazy.cpp	2009-11-24 12:30:45.000000000 +0100
@@ -131,7 +131,11 @@
 							break;
 							
 					case SDLK_s:
-							CORE->ScreenShot( "screenshots/Crazy",++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Crazy", getenv("HOME"));
+							CORE->ScreenShot( cfg_path,++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;	
 							
 					case SDLK_ESCAPE:
@@ -1427,8 +1431,12 @@
 				switch ( event.key.keysym.sym ) {
 							
 					// screenshot
-					case SDLK_s:							
-							CORE->ScreenShot("screenshots/Crazy",++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+					case SDLK_s:
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Crazy", getenv("HOME"));							
+							CORE->ScreenShot(cfg_path,++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;
 					default: ;
 				} // switch
diff -Naur mytoy-se-1.0.0/cdogcatcher.cpp mytoy-se-1.0.0-home/cdogcatcher.cpp
--- mytoy-se-1.0.0/cdogcatcher.cpp	2008-06-13 19:04:29.000000000 +0200
+++ mytoy-se-1.0.0-home/cdogcatcher.cpp	2009-11-24 12:30:45.000000000 +0100
@@ -115,7 +115,11 @@
 							break;
 							
 					case SDLK_s:
-							CORE->ScreenShot( "screenshots/Dog",++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Dog", getenv("HOME"));
+							CORE->ScreenShot( cfg_path,++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;	
 							
 					case SDLK_ESCAPE:
diff -Naur mytoy-se-1.0.0/cmenu.cpp mytoy-se-1.0.0-home/cmenu.cpp
--- mytoy-se-1.0.0/cmenu.cpp	2005-02-21 21:16:18.000000000 +0100
+++ mytoy-se-1.0.0-home/cmenu.cpp	2009-11-24 12:30:45.000000000 +0100
@@ -131,7 +131,11 @@
 							
 					// képernyõ mentése		
 					case SDLK_s:
-							CORE->ScreenShot( "screenshots/Menu",scrshotnr++,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Menu", getenv("HOME"));
+							CORE->ScreenShot( cfg_path,scrshotnr++,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;
 						
 					default:
diff -Naur mytoy-se-1.0.0/cmeteorfly.cpp mytoy-se-1.0.0-home/cmeteorfly.cpp
--- mytoy-se-1.0.0/cmeteorfly.cpp	2008-06-13 19:02:00.000000000 +0200
+++ mytoy-se-1.0.0-home/cmeteorfly.cpp	2009-11-24 12:30:45.000000000 +0100
@@ -130,7 +130,11 @@
 							break;
 							
 					case SDLK_s:
-							CORE->ScreenShot( "screenshots/Meteor",++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Meteor", getenv("HOME"));
+							CORE->ScreenShot( cfg_path,++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;	
 							
 					case SDLK_ESCAPE:
diff -Naur mytoy-se-1.0.0/configuration mytoy-se-1.0.0-home/configuration
--- mytoy-se-1.0.0/configuration	2005-02-21 21:16:30.000000000 +0100
+++ mytoy-se-1.0.0-home/configuration	2009-11-24 12:32:21.000000000 +0100
@@ -12,3 +12,4 @@
 time_stamp_period=1000
 vertical_mirror=1
 horizontal_mirror=1
+video_device=0
diff -Naur mytoy-se-1.0.0/coreheaders.h mytoy-se-1.0.0-home/coreheaders.h
--- mytoy-se-1.0.0/coreheaders.h	2008-06-13 19:17:59.000000000 +0200
+++ mytoy-se-1.0.0-home/coreheaders.h	2009-11-24 12:30:45.000000000 +0100
@@ -32,6 +32,10 @@
  
 */
 
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+
 #ifdef _WIN32
 #include <windows.h>
 #endif
diff -Naur mytoy-se-1.0.0/cwindow.cpp mytoy-se-1.0.0-home/cwindow.cpp
--- mytoy-se-1.0.0/cwindow.cpp	2008-06-13 19:02:19.000000000 +0200
+++ mytoy-se-1.0.0-home/cwindow.cpp	2009-11-24 12:30:45.000000000 +0100
@@ -91,7 +91,11 @@
 							break;
 							
 					case SDLK_s:
-							CORE->ScreenShot( "screenshots/Window",++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
+							char cfg_path[300];
+							sprintf(cfg_path, "%s/.mytoyse", getenv("HOME"));
+							mkdir(cfg_path, 0755);
+							sprintf(cfg_path, "%s/.mytoyse/Window", getenv("HOME"));
+							CORE->ScreenShot( cfg_path,++scrshotnr,0,0,GL_SCREEN_WIDTH,GL_SCREEN_HEIGHT);
 							break;	
 							
 					case SDLK_ESCAPE:
diff -Naur mytoy-se-1.0.0/mytoy_se.cpp mytoy-se-1.0.0-home/mytoy_se.cpp
--- mytoy-se-1.0.0/mytoy_se.cpp	2005-02-21 21:16:18.000000000 +0100
+++ mytoy-se-1.0.0-home/mytoy_se.cpp	2009-11-24 12:32:21.000000000 +0100
@@ -74,7 +74,7 @@
 	}
 	
 	// 2. Core "inicializálása"
-	if ( (started = GameCore->DetectCamera() ) ) { // Kamera detektálása
+//	if ( (started = GameCore->DetectCamera() ) ) { // Kamera detektálása
 		
 		if ( (started = GameCore->CreateSDLGLWindow() ) ) {  // SDLGL ablak létrehozása							
 			
@@ -116,7 +116,7 @@
 			// 3/b. Menu (játék) indítása
 			Menu->Start();							
 		}			
-	}
+//	}
 	
 	// ha valmilyen okból (kameradetektálás, OpenGL ablak) nem sikerült elindítani a CORE-t
 	if ( !started )
