diff -Naur mytoy-se-1.0.0/ccrazy.cpp mytoy-se-1.0.0-home-best/ccrazy.cpp
--- mytoy-se-1.0.0/ccrazy.cpp	2009-11-24 12:30:45.000000000 +0100
+++ mytoy-se-1.0.0-home-best/ccrazy.cpp	2009-11-24 13:45:24.000000000 +0100
@@ -179,7 +179,11 @@
 						
 				// legjobb fotója
 				case 20:
-						MakeTheBestPhoto("data/crazybubbles/pictures/the_best");						
+						char best_path[300];
+						sprintf(best_path, "%s/.mytoyse", getenv("HOME"));
+						mkdir(best_path, 0755);
+						sprintf(best_path, "%s/.mytoyse/crazybubbles-the_best", getenv("HOME"));
+						MakeTheBestPhoto(best_path);
 						gamestatus = 0;
 						
 						break;
@@ -225,6 +229,8 @@
 // Textúrák, hangok betöltése
 // ===========================
 int CCrazy::LoadItems() {
+	char best_path[300];
+	sprintf(best_path, "%s/.mytoyse/crazybubbles-the_best_0.png", getenv("HOME"));
 
 	// panel
 	if ( !(TexPanel = CORE->TEXMAN->CreateTexture("data/crazybubbles/pictures/panel.png") ) ) {
@@ -259,7 +265,7 @@
 	}
 	
 	// Legjobb fényképének, pontszámának betöltése (512*512*3)			
-	if ( !(TexTheBest = LoadTheBestPlayer("data/crazybubbles/pictures/the_best_0.png") ) )
+	if ( !(TexTheBest = LoadTheBestPlayer(best_path) ) )
 		if ( !(TexTheBest = CORE->TEXMAN->CreateTexture("data/misc/pictures/thebest_default.png") ) )
 		 	if ( !(TexTheBest = CORE->TEXMAN->CreateEmptyTexture(512,512,3) ) ) {
 		
diff -Naur mytoy-se-1.0.0/cdogcatcher.cpp mytoy-se-1.0.0-home-best/cdogcatcher.cpp
--- mytoy-se-1.0.0/cdogcatcher.cpp	2009-11-24 12:30:45.000000000 +0100
+++ mytoy-se-1.0.0-home-best/cdogcatcher.cpp	2009-11-24 13:44:36.000000000 +0100
@@ -173,7 +173,11 @@
 				
 				// Legjobb játékos fotója
 				case 20:
-						MakeTheBestPhoto("data/dogcatcher/pictures/the_best");
+						char best_path[300];
+						sprintf(best_path, "%s/.mytoyse", getenv("HOME"));
+						mkdir(best_path, 0755);
+						sprintf(best_path, "%s/.mytoyse/dogcatcher-the_best", getenv("HOME"));
+						MakeTheBestPhoto(best_path);
 						gamestatus = 0;
 						
 						break;
@@ -216,6 +220,8 @@
 // =======================
 
 int CDogCatcher::LoadItems() {
+	char best_path[300];
+	sprintf(best_path, "%s/.mytoyse/dogcatcher-the_best_0.png", getenv("HOME"));
 
 	// Menü képe	
 	if ( !(TexDogMenu = CORE->TEXMAN->CreateTexture("data/dogcatcher/pictures/dogcatchmenu.png") ) ) {
@@ -287,7 +293,7 @@
 	}
 		
 	// Legjobb fényképének, pontszámának betöltése (512*512*3)			
-	if ( !(TexTheBest = LoadTheBestPlayer("data/dogcatcher/pictures/the_best_0.png") ) )
+	if ( !(TexTheBest = LoadTheBestPlayer(best_path) ) )
 		if ( !(TexTheBest = CORE->TEXMAN->CreateTexture("data/misc/pictures/thebest_default.png") ) )
 		 	if ( !(TexTheBest = CORE->TEXMAN->CreateEmptyTexture(512,512,3) ) ) {
 		
diff -Naur mytoy-se-1.0.0/cmeteorfly.cpp mytoy-se-1.0.0-home-best/cmeteorfly.cpp
--- mytoy-se-1.0.0/cmeteorfly.cpp	2009-11-24 12:30:45.000000000 +0100
+++ mytoy-se-1.0.0-home-best/cmeteorfly.cpp	2009-11-24 13:45:47.000000000 +0100
@@ -187,7 +187,11 @@
 						
 				// legjobb fotója
 				case 20:
-						MakeTheBestPhoto("data/meteorfly/pictures/the_best");						
+						char best_path[300];
+						sprintf(best_path, "%s/.mytoyse", getenv("HOME"));
+						mkdir(best_path, 0755);
+						sprintf(best_path, "%s/.mytoyse/meteorfly-the_best", getenv("HOME"));
+						MakeTheBestPhoto(best_path);						
 						gamestatus = 0;
 						
 						break;
@@ -235,7 +239,8 @@
 // Meteorfly hangok, képek betöltése
 // ====================================
 int CMeteorfly::LoadItems() {	
-	
+	char best_path[300];
+	sprintf(best_path, "%s/.mytoyse/meteorfly-the_best_0.png", getenv("HOME"));
 	
 	// panel
 	if ( !(TexPanel = CORE->TEXMAN->CreateTexture("data/meteorfly/pictures/panel.png") ) ) {
@@ -293,8 +298,8 @@
 		return 0;
 	}
 	
-	// Legjobb fényképének, pontszámának betöltése (512*512*3)			
-	if ( !(TexTheBest = LoadTheBestPlayer("data/meteorfly/pictures/the_best_0.png") ) )
+	// Legjobb fényképének, pontszámának betöltése (512*512*3)		
+	if ( !(TexTheBest = LoadTheBestPlayer(best_path) ) )
 		if ( !(TexTheBest = CORE->TEXMAN->CreateTexture("data/misc/pictures/thebest_default.png") ) )
 		 	if ( !(TexTheBest = CORE->TEXMAN->CreateEmptyTexture(512,512,3) ) ) {
 		
diff -Naur mytoy-se-1.0.0/cwindow.cpp mytoy-se-1.0.0-home-best/cwindow.cpp
--- mytoy-se-1.0.0/cwindow.cpp	2009-11-24 12:30:45.000000000 +0100
+++ mytoy-se-1.0.0-home-best/cwindow.cpp	2009-11-24 13:46:04.000000000 +0100
@@ -159,7 +159,11 @@
 										
 				// Legjobb játékos fotója
 				case 20:
-						MakeTheBestPhoto("data/windowcleaner/pictures/the_best");
+						char best_path[300];
+						sprintf(best_path, "%s/.mytoyse", getenv("HOME"));
+						mkdir(best_path, 0755);
+						sprintf(best_path, "%s/.mytoyse/windowcleaner-the_best", getenv("HOME"));
+						MakeTheBestPhoto(best_path);
 						gamestatus = 0;
 						#ifdef PLAY_SOUNDS		
 							windowmenumusic.PlayMusic(-1);		
@@ -202,7 +206,8 @@
 // Képek, hangok betöltése
 // =======================
 int CWindow::LoadItems() {
-	
+	char best_path[300];
+	sprintf(best_path, "%s/.mytoyse/windowcleaner-the_best_0.png", getenv("HOME"));
 	
 	if ( !(TexPanel = CORE->TEXMAN->CreateTexture("data/windowcleaner/pictures/panel.png") ) ) {	
 		printf("I can't load the Panel!\n");
@@ -246,7 +251,7 @@
 	}
 		
 	// Legjobb fényképének, pontszámának betöltése (512*512*3)			
-	if ( !(TexTheBest = LoadTheBestPlayer("data/windowcleaner/pictures/the_best_0.png") ) )
+	if ( !(TexTheBest = LoadTheBestPlayer(best_path) ) )
 		if ( !(TexTheBest = CORE->TEXMAN->CreateTexture("data/misc/pictures/thebest_default.png") ) )
 		 	if ( !(TexTheBest = CORE->TEXMAN->CreateEmptyTexture(512,512,3) ) ) {
 		
