diff --git a/simgear/scene/tgdb/pt_lights.cxx b/simgear/scene/tgdb/pt_lights.cxx
index 5b7cf37..0e7a557 100644
--- a/simgear/scene/tgdb/pt_lights.cxx
+++ b/simgear/scene/tgdb/pt_lights.cxx
@@ -207,6 +207,7 @@ Effect* getLightEffect(float size, const Vec3& attenuation,
     std::vector<std::string> combinedExtensions;
     combinedExtensions.push_back("GL_ARB_point_sprite");
     combinedExtensions.push_back("GL_ARB_point_parameters");
+#if 0
     Technique* combinedTniq = new Technique;
     combinedTniq->passes.push_back(combinedPass);
     combinedTniq->setGLExtensionsPred(2.0, combinedExtensions);
@@ -217,6 +218,7 @@ Effect* getLightEffect(float size, const Vec3& attenuation,
     spriteTniq->passes.push_back(spritePass);
     spriteTniq->setGLExtensionsPred(2.0, spriteExtensions);
     effect->techniques.push_back(spriteTniq);
+#endif
     std::vector<std::string> parameterExtensions;
     parameterExtensions.push_back(combinedExtensions.back());
     Technique* parameterTniq = new Technique;
