diff -Naur src/character/rule_base.h src-diff/character/rule_base.h
--- src/character/rule_base.h	2008-12-31 18:41:14.000000000 +0100
+++ src-diff/character/rule_base.h	2009-04-17 14:10:31.000000000 +0200
@@ -68,7 +68,7 @@
     ~RuleBase()
     {
       g_round2 = false;
-      g_rand.reset();
+//      g_rand.reset();
     }
 
     float getDrawPriority() { return 10.0f; }
diff -Naur src/game.h src-diff/game.h
--- src/game.h	2009-01-04 01:05:04.000000000 +0100
+++ src-diff/game.h	2009-04-17 14:10:12.000000000 +0200
@@ -82,7 +82,9 @@
       
     void save()
     {
-      FILE *f = fopen("config", "wb");
+      char path[PATH_MAX];
+      snprintf(path, PATH_MAX, "%s/.exception_conflict/config", getenv("HOME"));
+      FILE *f = fopen(path, "wb");
       if(!f) {
         return;
       }
@@ -143,7 +145,9 @@
 
     void load()
     {
-      FILE *in = fopen("config", "rb");
+      char path[PATH_MAX];
+      snprintf(path, PATH_MAX, "%s/.exception_conflict/config", getenv("HOME"));
+      FILE *in = fopen(path, "rb");
       if(!in) {
         return;
       }
diff -Naur src/lib/sgui/sgui.h src-diff/lib/sgui/sgui.h
--- src/lib/sgui/sgui.h	2009-01-04 13:33:14.000000000 +0100
+++ src-diff/lib/sgui/sgui.h	2009-04-17 14:21:29.000000000 +0200
@@ -23,7 +23,7 @@
   #include <dirent.h>
 #endif
 #include <boost/thread.hpp>
-#include <FTGL/FTGLTextureFont.h>
+#include <FTGL/ftgl.h>
 #include <SDL/SDL.h>
 #include "sgui.h"
 
diff -Naur src/main.cc src-diff/main.cc
--- src/main.cc	2009-01-01 19:13:42.000000000 +0100
+++ src-diff/main.cc	2009-04-17 14:41:33.000000000 +0200
@@ -56,6 +56,9 @@
   char *hoge = new char[128];
 #endif
 
+  char cfg_path[PATH_MAX];
+  snprintf(cfg_path, PATH_MAX, "%s/.exception_conflict/", getenv("HOME"));
+  mkdir(cfg_path, 0755);
 
 #ifdef WIN32
   while(FindProcess("updater.exe")) {
diff -Naur src/Rakefile src-diff/Rakefile
--- src/Rakefile	2009-01-01 19:13:42.000000000 +0100
+++ src-diff/Rakefile	2009-04-17 14:26:47.000000000 +0200
@@ -4,7 +4,7 @@
 
 CFLAGS = "-O2 -I ./ -I lib #{`freetype-config --cflags`} #{`sdl-config --cflags`}".gsub(/\n/, '')
 LFLAGS = ''
-LIBS = "-L lib/ist -L lib/sgui -lsgui -list -lboost_regex-gcc41-mt -lboost_date_time-gcc41-mt -lboost_system-gcc41-mt -lboost_thread-gcc41-mt -lz -lpng -lGL -lGLU -lGLEW -lftgl #{`sdl-config --libs`} -lSDL_mixer #{`freetype-config --libs`}".gsub(/\n/, '')
+LIBS = "-L lib/ist -L lib/sgui -lsgui -list -lboost_regex-mt -lboost_date_time-mt -lboost_system-mt -lboost_thread-mt -lz -lpng -lGL -lGLU -lGLEW -lftgl #{`sdl-config --libs`} -lSDL_mixer #{`freetype-config --libs`}".gsub(/\n/, '')
 CC = 'g++'
 
 SRCS = Dir.glob("{.,character}/*.cc")
@@ -25,11 +25,10 @@
 
 
 task "default" => "all"
-task "all" => ["ist", "sgui", "exception"]
+task "all" => ["ist", "sgui", "exception_conflict"]
 
-file "exception" => OBJS do |t|
+file "exception_conflict" => OBJS do |t|
   sh "#{CC} -o #{t.name} #{t.prerequisites.join(' ')} #{CFLAGS} #{LIBS}"
-  sh "cp #{t.name} ../#{t.name}"
 end
 
 rule ".o" => ".cc" do |t|
