--- redeclipse-1.5.2.orig/src/Makefile	2015-07-11 14:20:26.607791059 +0500
+++ redeclipse-1.5.2/src/Makefile	2015-07-11 14:37:20.846741852 +0500
@@ -37,7 +37,7 @@
 CXX_TEMP:=$(CXX)
 override CXX=$(TOOLSET_PREFIX)$(CXX_TEMP)
 
-INCLUDES= -I. -Ishared -Iengine -Igame -Ienet/include
+INCLUDES= -I. -Ishared -Iengine -Igame
 
 
 STRIP=
@@ -81,7 +81,7 @@
 endif
 endif
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl-config --cflags`
-CLIENT_LIBS= -Lenet -lenet -L/usr/X11R6/lib -lX11 `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
+CLIENT_LIBS= -lenet -L/usr/X11R6/lib -lX11 `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
 endif
 ifneq (,$(findstring linux,$(PLATFORM)))
 CLIENT_LIBS+= -lrt
@@ -156,7 +156,7 @@
 SERVER_LIBS= -mwindows $(STD_LIBS) -L$(WINDLL) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -Lenet -lenet -lz
+SERVER_LIBS= -lenet -lz
 endif
 SERVER_OBJS= \
 	shared/crypto-standalone.o \
@@ -170,17 +170,12 @@
 	engine/server-standalone.o \
 	game/server-standalone.o
 
-LIBENET= enet/libenet.a
-
 GENKEY_OBJS= engine/genkey.o shared/crypto-standalone.o
 
 all:
 
 default: all
 
-clean-enet:
-	$(MAKE) -C enet clean
-
 clean-client:
 	@rm -fv $(CLIENT_PCH) $(CLIENT_OBJS) $(APPCLIENT)$(BIN_SUFFIX)
 
@@ -190,7 +185,7 @@
 clean-genkey:
 	@rm -fv $(GENKEY_OBJS) genkey$(BIN_SUFFIX)
 
-clean: clean-enet clean-client clean-server clean-genkey
+clean: clean-client clean-server clean-genkey
 
 %.h.gch: %.h
 	$(CXX) $(CXXFLAGS) -x c++-header -o $(subst .h.gch,.tmp.h.gch,$@) $(subst .h.gch,.h,$@)
@@ -256,9 +251,6 @@
 
 install: install-client install-server
 
-$(LIBENET):
-	$(MAKE) -C enet
-
 depend:
 	makedepend -Y -I. -Ishared -Iengine -Igame $(subst .o,.cpp,$(CLIENT_OBJS))
 	makedepend -a -o-standalone.o -Y -I. -Ishared -Iengine -Igame -DSTANDALONE $(subst -standalone.o,.cpp,$(SERVER_OBJS))
--- redeclipse-1.5.2.orig/src/engine/engine.h	2015-07-11 14:20:25.554791110 +0500
+++ redeclipse-1.5.2/src/engine/engine.h	2015-07-11 15:01:02.790672864 +0500
@@ -3,6 +3,7 @@
 
 #include "version.h"
 #include "cube.h"
+#include <enet/enet.h>
 
 #define LAN_PORT 28799
 #define MASTER_PORT 28800
