--- configure.ac.old	2010-11-11 00:34:07.000000000 +0100
+++ configure.ac	2010-11-11 00:37:20.000000000 +0100
@@ -745,23 +745,36 @@
 fi
 
 dnl Check for Subversion library support
+# defaults to no
 save_LIBS=$LIBS
 save_CPPFLAGS=$CPPFLAGS
-LIBS="`apr-1-config --link-ld`"
-CPPFLAGS="-I/usr/include/subversion-1 `apr-1-config --includes`"
-AC_CHECK_HEADERS([svn_client.h])
-if test "x$ac_cv_header_svn_client_h" != "xyes"; then
-  echo "TerraSync will shell out for command line subversion"
+AC_ARG_WITH(subversion, [  --with-subversion     Link libsvn against terrasync [default=no]], [], [with_subversion=no])
+if test "x$with_subversion" = "xyes"; then
+    LIBS="`apr-1-config --link-ld`"
+    CPPFLAGS="-I/usr/include/subversion-1 `apr-1-config --includes`"
+    AC_CHECK_HEADERS([svn_client.h])
+       if test "x$ac_cv_header_svn_client_h" = "xyes"; then
+       echo "TerraSync will use integrated subversion library"
+       AC_SEARCH_LIBS(svn_client_checkout, svn_client-1)
+       AC_SEARCH_LIBS(svn_cmdline_init, svn_subr-1)
+       svn_LIBS=$LIBS
+       svn_CPPFLAGS=$CPPFLAGS
+       AC_SUBST(svn_LIBS)
+       AC_SUBST(svn_CPPFLAGS)
+       else
+       echo
+       echo "You requested libsvn support, but the library could not be found"
+       echo "on your system"
+       echo
+       echo "Please either install it or compile without Subversion support"
+       echo
+       echo "configure aborted."
+       exit
+       fi
+else
+echo "TerraSync will shell out for command line subversion"
   svn_LIBS=""
   svn_CPPFLAGS=""
-else
-  echo "TerraSync will use integrated subversion library"
-  AC_SEARCH_LIBS(svn_client_checkout, svn_client-1)
-  AC_SEARCH_LIBS(svn_cmdline_init, svn_subr-1)
-  svn_LIBS=$LIBS
-  svn_CPPFLAGS=$CPPFLAGS
-  AC_SUBST(svn_LIBS)
-  AC_SUBST(svn_CPPFLAGS)
 fi
 LIBS=$save_LIBS
 CPPFLAGS=$save_CPPFLAGS
