--- usr/bin/steam_orig	2013-01-10 11:35:42.777775828 +0400
+++ usr/bin/steam	2013-01-10 11:37:02.936776107 +0400
@@ -40,7 +40,7 @@
 		# Save the prompt in a temporary file because it can have newlines in it
 		tmpfile="$(mktemp || echo "/tmp/steam_message.txt")"
 		echo -e "$*" >"$tmpfile"
-		xterm -T "$title" -e "cat $tmpfile; echo -n 'Press enter to continue: '; read input"
+		"${TERM}" -T "$title" -e "cat $tmpfile; echo -n 'Press enter to continue: '; read input"
 		rm -f "$tmpfile"
 	fi
 }
